From 76a8364e0ea7de2f1b30e88eec21e29a4e2e516c Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Mon, 8 Oct 2018 21:14:05 +1100
Subject: [PATCH] meson: don't build tests or the thumbnailer if cross building

---
 meson.build | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 6b0ba49..56ef4f1 100644
--- a/meson.build
+++ b/meson.build
@@ -396,8 +396,10 @@ endif
 # i18n
 subdir('po')
 
-subdir('tests')
-subdir('thumbnailer')
+if not meson.is_cross_build()
+  subdir('tests')
+  subdir('thumbnailer')
+endif
 
 # Documentation
 subdir('docs')
-- 
2.19.0

