From 279c222e59caec8b315277938b56d3f3a51a319c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Wed, 22 Aug 2012 15:50:12 +0200
Subject: [PATCH 3/4] Link the other libs with -lm too

---
 lib/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index d095eab..87f35fc 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -20,11 +20,11 @@ libvorbis_la_LIBADD = @VORBIS_LIBS@ @OGG_LIBS@ -lm
 
 libvorbisfile_la_SOURCES = vorbisfile.c
 libvorbisfile_la_LDFLAGS = -no-undefined -version-info @VF_LIB_CURRENT@:@VF_LIB_REVISION@:@VF_LIB_AGE@
-libvorbisfile_la_LIBADD = libvorbis.la @OGG_LIBS@
+libvorbisfile_la_LIBADD = libvorbis.la @OGG_LIBS@ -lm
 
 libvorbisenc_la_SOURCES = vorbisenc.c 
 libvorbisenc_la_LDFLAGS = -no-undefined -version-info @VE_LIB_CURRENT@:@VE_LIB_REVISION@:@VE_LIB_AGE@
-libvorbisenc_la_LIBADD = libvorbis.la @OGG_LIBS@
+libvorbisenc_la_LIBADD = libvorbis.la @OGG_LIBS@ -lm
 
 EXTRA_PROGRAMS = barkmel tone psytune
 CLEANFILES = $(EXTRA_PROGRAMS)
-- 
1.8.4

