From e5fe93640e77dab1902ded867edd6b6acf80ad06 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Wed, 6 Nov 2013 10:48:00 +0100
Subject: [PATCH] Use our version of automake instead of some random older
 version

---
 autogen.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 0aca638..e605da8 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -47,8 +47,8 @@ if test -r Makefile.am; then
     echo -n "checking for automake $AM_NEEDED or later... "
     majneeded=`echo $AM_NEEDED | $VERSIONMKMAJ`
     minneeded=`echo $AM_NEEDED | $VERSIONMKMIN`
-    for am in automake-$AM_NEEDED automake$AM_NEEDED \
-	automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake; do
+    for am in automake automake-$AM_NEEDED automake$AM_NEEDED \
+	automake-1.10 automake-1.9 automake-1.8 automake-1.7; do
       ($am --version < /dev/null > /dev/null 2>&1) || continue
       ver=`$am --version < /dev/null | head -n 1 | $VERSIONGREP`
       maj=`echo $ver | $VERSIONMKMAJ`
@@ -61,8 +61,8 @@ if test -r Makefile.am; then
     done
     test -z $AUTOMAKE &&  echo "no"
     echo -n "checking for aclocal $AM_NEEDED or later... "
-    for ac in aclocal-$AM_NEEDED aclocal$AM_NEEDED \
-	aclocal-1.10 aclocal-1.9 aclocal-1.8 aclocal-1.7 aclocal; do
+    for ac in aclocal aclocal-$AM_NEEDED aclocal$AM_NEEDED \
+	aclocal-1.10 aclocal-1.9 aclocal-1.8 aclocal-1.7; do
       ($ac --version < /dev/null > /dev/null 2>&1) || continue
       ver=`$ac --version < /dev/null | head -n 1 | $VERSIONGREP`
       maj=`echo $ver | $VERSIONMKMAJ`
-- 
1.8.4.2

