From 37e880e9d13e81f87362f62835b22ce38ae73fe1 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 7 Mar 2012 18:36:35 +0100
Subject: [PATCH 01/14] Check properly for sched_setscheduler

---
 configure.ac         | 2 +-
 encodedv/dvconnect.c | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2b95735..8c3b2f6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -242,7 +242,7 @@ AC_HEADER_TIME
 
 dnl Checks for library functions.
 AC_FUNC_MMAP
-AC_CHECK_FUNCS(gettimeofday)
+AC_CHECK_FUNCS(gettimeofday, sched_setscheduler)
 
 AC_SUBST(ac_aux_dir)
 AC_SUBST(RPM_RELEASE)
diff --git a/encodedv/dvconnect.c b/encodedv/dvconnect.c
index f276846..8492dfe 100644
--- a/encodedv/dvconnect.c
+++ b/encodedv/dvconnect.c
@@ -859,6 +859,7 @@ int rt_raisepri (int pri)
 {
 #ifdef _SC_PRIORITY_SCHEDULING
 	struct sched_param scp;
+        int ret = -1;
 
 	/*
 	 * Verify that scheduling is available
@@ -870,7 +871,10 @@ int rt_raisepri (int pri)
 	} else {
 		memset (&scp, '\0', sizeof (scp));
 		scp.sched_priority = sched_get_priority_max (SCHED_RR) - pri;
-		if (sched_setscheduler (0, SCHED_RR, &scp) < 0)	{
+#ifdef HAVE_SCHED_SETSCHEDULER
+		ret = sched_setscheduler (0, SCHED_RR, &scp)
+#endif
+		if (ret < 0)	{
 			fprintf (stderr, "WARNING: Cannot set RR-scheduler\n");
 			return (-1);
 		}
-- 
2.1.4

