From a0d2365c7b70b5baefa2f6dd4822caac9037790e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Tue, 13 Nov 2012 16:25:31 +0100
Subject: [PATCH 03/14] Don't build test programs

---
 Makefile.am       | 2 +-
 libdv/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 659a7b2..831bb9b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@ else
 gtk_subdirs =
 endif
 
-SUBDIRS                   = libdv encodedv $(gtk_subdirs)
+SUBDIRS                   = libdv #encodedv $(gtk_subdirs)
 
 AUX_DIST                  = $(ac_aux_dir)/config.guess \
                                    $(ac_aux_dir)/config.sub \
diff --git a/libdv/Makefile.am b/libdv/Makefile.am
index 2b010a3..a837bdf 100644
--- a/libdv/Makefile.am
+++ b/libdv/Makefile.am
@@ -16,7 +16,7 @@ GASMOFF=gasmoff
 endif # HOST_X86_64
 
 
-noinst_PROGRAMS= dovlc testvlc testbitstream $(GASMOFF) recode reppm enctest
+noinst_PROGRAMS= #dovlc testvlc testbitstream $(GASMOFF) recode reppm enctest
 
 #
 # If HOST_X86 is set, we build all the x86 asm stuff..
-- 
2.1.4

