From 0e925478a129657c449798e151b51fdee8b476ed Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Wed, 14 Nov 2012 12:15:50 +0100
Subject: [PATCH 06/14] Add the pthread libs to pkg-config

---
 libdv.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdv.pc.in b/libdv.pc.in
index 2f1ecb9..a98f6e8 100644
--- a/libdv.pc.in
+++ b/libdv.pc.in
@@ -7,5 +7,5 @@ Name: libdv
 Description: DV software codec
 Requires: 
 Version: @VERSION@
-Libs: -L${libdir} -ldv -lm @REQUIRES_NOPKGCONFIG@
+Libs: -L${libdir} -ldv -lm @PTHREAD_LIBS@ @REQUIRES_NOPKGCONFIG@
 Cflags: -I${includedir}
-- 
2.1.4

