From 041cc9e9a878f8357781154efe4ebe9e7b701e86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Wed, 28 Nov 2012 11:51:26 +0100
Subject: [PATCH 12/14] Build a DLL on Windows

---
 libdv/Makefile.am | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libdv/Makefile.am b/libdv/Makefile.am
index c5955ad..05f23da 100644
--- a/libdv/Makefile.am
+++ b/libdv/Makefile.am
@@ -59,7 +59,7 @@ libdv_la_SOURCES= dv.c dct.c idct_248.c weighting.c quant.c vlc.c place.c \
         encode.c headers.c enc_input.c enc_audio_input.c enc_output.c \
 	$(libdv_la_ASMS)
 
-libdv_la_LDFLAGS = -version-info 4:3:0
+libdv_la_LDFLAGS = -version-info 4:3:0 -no-undefined
 libdv_la_LIBADD = $(PTHREAD_LIBS)
 
 dovlc_SOURCES= dovlc.c 
-- 
2.1.4

