From cfdd83e5ab8027e8dfe3c033a3db79df1079d6ba Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 10 Sep 2012 12:03:05 +0200
Subject: [PATCH 2/4] fix autoreconf

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7608315..4fcd48b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -26,7 +26,7 @@ AC_PREREQ(2.53)
 
 AC_CONFIG_SRCDIR([decoder.h])
 
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 
 AM_CONFIG_HEADER([config.h])
 
-- 
2.1.0

