From bf1379e0f3fb9419fcb18e4c665f5ddba4a30418 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Wed, 1 May 2024 16:51:06 +0000
Subject: [PATCH 4/4] Add missing libcroco dependency

---
 librsvg.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/librsvg.pc.in b/librsvg.pc.in
index a451de0..608807c 100644
--- a/librsvg.pc.in
+++ b/librsvg.pc.in
@@ -9,7 +9,7 @@ css_supported=true
 Name: librsvg
 Description: library that renders svg files
 Version: @VERSION@
-Requires: glib-2.0 gio-2.0 gdk-pixbuf-2.0 cairo
+Requires: glib-2.0 gio-2.0 gdk-pixbuf-2.0 cairo libcroco-0.6
 Requires.private:
 Libs: -L${libdir} -lrsvg-@RSVG_API_MAJOR_VERSION@ -lm
 Cflags: -I${includedir}/librsvg-@RSVG_API_VERSION@
-- 
2.44.0

