From f01a1e0dcc03cc65a6444a37f1da46477d6a0c24 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 1 Mar 2012 18:20:51 +0100
Subject: [PATCH 2/3] Add -no-undefined on windows

---
 configure.in        | 7 +++++++
 libfaad/Makefile.am | 4 ++++
 2 files changed, 11 insertions(+)

diff --git a/configure.in b/configure.in
index 6315e01..068dd68 100644
--- a/configure.in
+++ b/configure.in
@@ -154,6 +154,13 @@ if test x$WITHDRM = xyes; then
   AC_DEFINE(DRM_PS, 1, [Define if you want support for Digital Radio Mondiale (DRM) parametric stereo])
 fi
 
+case "$host" in
+  *-*-mingw*)
+    os_win32=yes
+  ;;
+esac
+AM_CONDITIONAL(OS_WIN32, [test "$os_win32" = "yes"])
+
 AC_CONFIG_FILES(libfaad/Makefile)
 AC_CONFIG_FILES(common/Makefile)
 AC_CONFIG_FILES(common/mp4ff/Makefile)
diff --git a/libfaad/Makefile.am b/libfaad/Makefile.am
index e5ad36d..81d7608 100644
--- a/libfaad/Makefile.am
+++ b/libfaad/Makefile.am
@@ -7,6 +7,10 @@ include_HEADERS = $(top_srcdir)/include/faad.h \
 libfaad_la_LDFLAGS = -version-info 2:0:0
 libfaad_la_LIBADD = -lm
 
+if OS_WIN32
+libfaad_la_LDFLAGS += -no-undefined
+endif
+
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
 		     drm_dec.c error.c filtbank.c \
 		     ic_predict.c is.c lt_predict.c mdct.c mp4.c ms.c output.c pns.c \
-- 
2.1.4

