From ba7e55f9d5f072889ba10132cde0eeac814507e2 Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Sat, 14 Feb 2015 20:55:24 +0100
Subject: [PATCH] Remove -no-undefined from LDFLAGS

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index fc51921..462bbf7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -73,13 +73,13 @@ libfl_la_SOURCES = \
 	libmain.c \
 	libyywrap.c
 
-libfl_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
+libfl_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
 
 libfl_pic_la_SOURCES = \
 	libmain.c \
 	libyywrap.c
 
-libfl_pic_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
+libfl_pic_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
 
 noinst_HEADERS = \
 	flexdef.h \
-- 
2.1.0

