From 54e3f4d658c089f555ab0848a73b99e302581177 Mon Sep 17 00:00:00 2001
From: Justin Kim <justin.kim@collabora.com>
Date: Tue, 15 Sep 2015 22:48:07 +0900
Subject: [PATCH] flex: prevent to build doc

---
 Makefile.am  | 1 -
 configure.ac | 1 -
 2 files changed, 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 1540bb7..c839eb1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -45,7 +45,6 @@ EXTRA_DIST = \
 SUBDIRS = \
 	lib \
 	src \
-	doc \
 	examples \
 	po \
 	tests \
diff --git a/configure.ac b/configure.ac
index c786752..c9b2151 100644
--- a/configure.ac
+++ b/configure.ac
@@ -146,7 +146,6 @@ AC_REPLACE_FUNCS(reallocarray)
 
 AC_CONFIG_FILES(
 Makefile
-doc/Makefile
 examples/Makefile
 examples/fastwc/Makefile
 examples/manual/Makefile
-- 
2.11.0

