From 340bead857cba9bbcd190650de4e17e5c7b85019 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 12 Oct 2018 18:12:47 +0530
Subject: [PATCH] meson: Don't run help2man when cross building

---
 utils/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/utils/meson.build b/utils/meson.build
index 3adcc32..510d1b7 100644
--- a/utils/meson.build
+++ b/utils/meson.build
@@ -40,7 +40,9 @@ pango_view = executable('pango-view', pango_view_sources,
                         ])
 
 help2man = find_program('help2man', required: false)
-if help2man.found()
+# help2man has to run pango-view to generate the man page, which it can't do if
+# we're cross building. We also can't tell help2man what exe wrapper to use.
+if help2man.found() and not meson.is_cross_build()
   help2man_opts = [
     '--no-info',
     '--section=1',
-- 
2.17.1

