From 8092554e6e14be784267a3eb829a804092248a7d Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@gnome.org>
Date: Wed, 2 Apr 2014 10:09:29 +0200
Subject: [PATCH] Do not build tests on windows

---
 Makefile.am | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 2b4a5b8..4262789 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,7 +23,12 @@
 
 SUBDIRS=fontconfig fc-blanks fc-case fc-lang fc-glyphname src \
 	fc-cache fc-cat fc-list fc-match fc-pattern fc-query fc-scan \
-	fc-validate conf.d test
+	fc-validate conf.d
+
+if !OS_WIN32
+SUBDIRS += test
+endif
+
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
-- 
2.11.0

