From dfeb55ef3e21b562f02a857af5d7203d9c6d3754 Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@gnome.org>
Date: Wed, 2 Apr 2014 10:09:29 +0200
Subject: [PATCH] Do not build tests on windows

---
 Makefile.am | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 2b949e4..4502bf6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,11 +23,15 @@
 
 SUBDIRS=fontconfig fc-case fc-lang fc-glyphname src \
 	fc-cache fc-cat fc-list fc-match fc-pattern fc-query fc-scan \
-	fc-validate conf.d test
+	fc-validate conf.d
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
 
+if !OS_WIN32
+SUBDIRS += test
+endif
+
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = \
-- 
1.9.0

