From 4fe8cbc9b86f5d7c2d12e0c3f84e97c42e9026b9 Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Sun, 25 May 2014 11:20:34 +0200
Subject: [PATCH 3/3] configure: Allow static build

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1086a9a..135e95a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,7 @@ fi
 
 dnl Initialize libtool
 LT_PREREQ([2.2])
-LT_INIT([disable-static win32-dll])
+LT_INIT([win32-dll])
 
 dnl libtool versioning
 
-- 
1.9.0

