From cafa81de8f9a61e3aa0db9e041a39a1941f9d63d Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 28 Mar 2014 15:25:53 +0100
Subject: [PATCH] osx: fix compiler error with newer clang

---
 gobject/gtype.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gobject/gtype.h b/gobject/gtype.h
index ff57af7..1400b05 100644
--- a/gobject/gtype.h
+++ b/gobject/gtype.h
@@ -1674,7 +1674,7 @@ static gint     TypeName##_private_offset; \
 \
 _G_DEFINE_TYPE_EXTENDED_CLASS_INIT(TypeName, type_name) \
 \
-static inline gpointer \
+G_GNUC_UNUSED static inline gpointer \
 type_name##_get_instance_private (TypeName *self) \
 { \
   return (G_STRUCT_MEMBER_P (self, TypeName##_private_offset)); \
-- 
1.8.3.2

