From 4c4448debd314c8069d7c6af53144aaab089ffd2 Mon Sep 17 00:00:00 2001
From: Hernan Martinez <hernan.c.martinez@gmail.com>
Date: Thu, 26 Jan 2023 10:23:41 +0000
Subject: [PATCH] Fix safe_wspawnve #define

---
 glib/gspawn-win32.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gspawn-win32.c b/glib/gspawn-win32.c
index 665d7313e..e8bc10e9a 100644
--- a/glib/gspawn-win32.c
+++ b/glib/gspawn-win32.c
@@ -163,7 +163,7 @@ safe_wspawnvpe (int _Mode,
 
 #else
 
-#define safe_wspawnve _spawnve
+#define safe_wspawnve _wspawnve
 #define safe_wspawnvpe _wspawnvpe
 
 #endif /* _UCRT */
-- 
2.44.0.windows.1

