From c627fb492624bdbbb9206d88e7852bddeacfcb00 Mon Sep 17 00:00:00 2001
From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Wed, 23 May 2018 10:07:22 -0400
Subject: [PATCH 3/3] Fix missing osx files when running "make dist"

Fixes #528, #1381.
---
 gio/Makefile.am | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gio/Makefile.am b/gio/Makefile.am
index bf2b40fd0..8eaf22e0a 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -601,13 +601,13 @@ libgio_2_0_la_LDFLAGS = $(GLIB_LINK_FLAGS) \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-dynamic $(no_undefined)
 
-if OS_COCOA
 # We cannot build .m files because it would require adding AC_PROG_OBJC in
 # configure.ac, and it cannot be added conditionally. That means we would always
 # depend on an objective-c compiler even when not needed. To work around that
 # limitation we rename .m files to .c and compile them separately with
 # -xobjective-c into an intermediary library. Note that we cannot rename those
 # files to .c directly in the source tree because Meson needs the .m extension.
+# This must be done outside of "if OS_COCOA" block otherwise distcheck fails.
 # See https://bugzilla.gnome.org/show_bug.cgi?id=672777.
 OBJC_FILES = \
 	gnextstepsettingsbackend.m \
@@ -618,9 +618,11 @@ OBJC_FILES = \
 OBJC_C_FILES = $(OBJC_FILES:.m=.c)
 BUILT_SOURCES += $(OBJC_C_FILES)
 CLEANFILES += $(OBJC_C_FILES)
+EXTRA_DIST += $(OBJC_FILES)
 $(OBJC_C_FILES): %.c: %.m Makefile
 	cp $< $@
 
+if OS_COCOA
 libgio_objc_2_0_la_CFLAGS = $(libgio_2_0_la_CFLAGS) -xobjective-c
 libgio_objc_2_0_la_CPPFLAGS = $(libgio_2_0_la_CPPFLAGS)
 libgio_objc_2_0_la_LDFLAGS = $(libgio_2_0_la_LDFLAGS) -Wl,-framework,Foundation -Wl,-framework,AppKit
-- 
2.17.1

