From 27df2bc8cd9d52f0e52ced942e244b3c9fbe48e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Fri, 4 Sep 2015 11:58:57 +0300
Subject: [PATCH] gioerror: Add more mappings for WinSock error codes

---
 gio/gioerror.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/gio/gioerror.c b/gio/gioerror.c
index 0faff25..c069347 100644
--- a/gio/gioerror.c
+++ b/gio/gioerror.c
@@ -301,6 +301,7 @@ g_io_error_from_win32_error (gint error_code)
 
     case WSA_INVALID_HANDLE:
     case WSA_INVALID_PARAMETER:
+    case WSAEINVAL:
     case WSAEBADF:
     case WSAENOTSOCK:
       return G_IO_ERROR_INVALID_ARGUMENT;
@@ -320,6 +321,19 @@ g_io_error_from_win32_error (gint error_code)
     case WSAECONNRESET:
       return G_IO_ERROR_CONNECTION_CLOSED;
 
+    case WSAEHOSTUNREACH:
+      return G_IO_ERROR_HOST_UNREACHABLE;
+
+    case WSAENETUNREACH:
+      return G_IO_ERROR_NETWORK_UNREACHABLE;
+
+    case WSAECONNREFUSED:
+      return G_IO_ERROR_CONNECTION_REFUSED;
+
+    case WSAETIMEDOUT:
+      return G_IO_ERROR_TIMED_OUT;
+
+    case WSAENOTCONN:
     case ERROR_PIPE_LISTENING:
       return G_IO_ERROR_NOT_CONNECTED;
 
-- 
2.5.1

