From 84b12eebc88e9fd161edda0ec679813fcfdd5028 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sat, 4 Jan 2014 19:40:24 +0100
Subject: [PATCH] windows: fix deadlock in make install

---
 Makefile.am | 12 ------------
 Makefile.in | 10 ----------
 2 files changed, 22 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e5f3f56..befc61e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -63,17 +63,5 @@ gen-ChangeLog:
 	fi
 
 
-if HAVE_W32_SYSTEM
-install-data-hook:
-	set -e; \
-	for i in $$($(top_srcdir)/potomo --get-linguas $(top_srcdir)/po); do \
-           $(MKDIR_P) "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES" || true; \
-           rm -f "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES/libgpg-error.mo" \
-                                                 2>/dev/null || true; \
-           $(top_srcdir)/potomo $(top_srcdir)/po/$$i.po \
-              "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES/libgpg-error.mo" ; \
-	done
-endif
-
 stowinstall:
 	$(MAKE) $(AM_MAKEFLAGS) install prefix=/usr/local/stow/libgpg-error
diff --git a/Makefile.in b/Makefile.in
index aa92e4d..f60fd45 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -901,16 +901,6 @@ gen-ChangeLog:
 	  mv $(distdir)/cl-t $(distdir)/ChangeLog;			\
 	fi
 
-@HAVE_W32_SYSTEM_TRUE@install-data-hook:
-@HAVE_W32_SYSTEM_TRUE@	set -e; \
-@HAVE_W32_SYSTEM_TRUE@	for i in $$($(top_srcdir)/potomo --get-linguas $(top_srcdir)/po); do \
-@HAVE_W32_SYSTEM_TRUE@           $(MKDIR_P) "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES" || true; \
-@HAVE_W32_SYSTEM_TRUE@           rm -f "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES/libgpg-error.mo" \
-@HAVE_W32_SYSTEM_TRUE@                                                 2>/dev/null || true; \
-@HAVE_W32_SYSTEM_TRUE@           $(top_srcdir)/potomo $(top_srcdir)/po/$$i.po \
-@HAVE_W32_SYSTEM_TRUE@              "$(DESTDIR)$(localedir)/$$i/LC_MESSAGES/libgpg-error.mo" ; \
-@HAVE_W32_SYSTEM_TRUE@	done
-
 stowinstall:
 	$(MAKE) $(AM_MAKEFLAGS) install prefix=/usr/local/stow/libgpg-error
 
-- 
2.1.0

