From edb9254d7917fdca8cbaec2a0a2e6444d250230e Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 29 Jan 2019 16:16:17 +0530
Subject: [PATCH 1/2] qt: Print the full path of the `qmake` binary found

---
 mesonbuild/dependencies/ui.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/dependencies/ui.py b/mesonbuild/dependencies/ui.py
index 03c96f22..a44bef50 100644
--- a/mesonbuild/dependencies/ui.py
+++ b/mesonbuild/dependencies/ui.py
@@ -362,7 +362,7 @@ class QtBaseDependency(ExternalDependency):
             return
         self.version = re.search(self.qtver + '(\.\d+)+', stdo).group(0)
         # Query library path, header path, and binary path
-        mlog.log("Found qmake:", mlog.bold(self.qmake.get_name()), '(%s)' % self.version)
+        mlog.log("Found qmake:", mlog.bold(self.qmake.get_path()), '(%s)' % self.version)
         stdo = Popen_safe(self.qmake.get_command() + ['-query'])[1]
         qvars = {}
         for line in stdo.split('\n'):
-- 
2.18.0.windows.1

