From ce8af2e8bff54809de0d2217d91d3f61e52bd06b Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sat, 14 Apr 2012 06:06:05 +0200
Subject: [PATCH 03/14] Use env tools if they are set

---
 Makefile            | 12 +++++++++---
 Makefile-libbz2_dll |  1 -
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 9754ddf..dac6c72 100644
--- a/Makefile
+++ b/Makefile
@@ -15,9 +15,15 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
+ifndef CC
+  CC=gcc
+endif
+ifndef RANLIB
+  RANLIB=ranlib
+endif
+ifndef AR
+  AR=ar
+endif
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
diff --git a/Makefile-libbz2_dll b/Makefile-libbz2_dll
index 1707a90..805ccb0 100644
--- a/Makefile-libbz2_dll
+++ b/Makefile-libbz2_dll
@@ -22,7 +22,6 @@
 
 
 SHELL=/bin/sh
-CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
 CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
 
-- 
1.8.3.1

