From 34dee02fe8ff3b5d533ccb3def418bb407e6f8dd Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 16 Apr 2012 00:06:20 +0200
Subject: [PATCH 05/14] Install dll in the bin dir

---
 Makefile-libbz2_dll | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile-libbz2_dll b/Makefile-libbz2_dll
index 805ccb0..2810af4 100644
--- a/Makefile-libbz2_dll
+++ b/Makefile-libbz2_dll
@@ -40,7 +40,7 @@ clean:
 	rm -f $(OBJS) libbz2.dll
 
 install:
-	cp -f libbz2.dll $(PREFIX)/lib
+	cp -f libbz2.dll $(PREFIX)/bin
 
 blocksort.o: blocksort.c
 	$(CC) $(CFLAGS) -c blocksort.c
-- 
1.8.3.1

