From a15ee210c1325c8cf0325a7ade907a63de0c12a9 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 20 Apr 2012 20:06:18 +0200
Subject: [PATCH 07/14] Add variable for executable extansions on Windows

---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index f692a3e..36ede3d 100644
--- a/Makefile
+++ b/Makefile
@@ -81,14 +81,14 @@ install: bzip2 bzip2recover
 	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
 	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
 	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
-	cp -f bzip2 $(PREFIX)/bin/bzip2
-	cp -f bzip2 $(PREFIX)/bin/bunzip2
-	cp -f bzip2 $(PREFIX)/bin/bzcat
-	cp -f bzip2recover $(PREFIX)/bin/bzip2recover
-	chmod a+x $(PREFIX)/bin/bzip2
-	chmod a+x $(PREFIX)/bin/bunzip2
-	chmod a+x $(PREFIX)/bin/bzcat
-	chmod a+x $(PREFIX)/bin/bzip2recover
+	cp -f bzip2$(EXT) $(PREFIX)/bin/bzip2$(EXT)
+	cp -f bzip2$(EXT) $(PREFIX)/bin/bunzip2$(EXT)
+	cp -f bzip2$(EXT) $(PREFIX)/bin/bzcat$(EXT)
+	cp -f bzip2recover$(EXT) $(PREFIX)/bin/bzip2recover$(EXT)
+	chmod a+x $(PREFIX)/bin/bzip2$(EXT)
+	chmod a+x $(PREFIX)/bin/bunzip2$(EXT)
+	chmod a+x $(PREFIX)/bin/bzcat$(EXT)
+	chmod a+x $(PREFIX)/bin/bzip2recover$(EXT)
 	cp -f bzip2.1 $(PREFIX)/man/man1
 	chmod a+r $(PREFIX)/man/man1/bzip2.1
 	cp -f bzlib.h $(PREFIX)/include
-- 
1.8.3.1

