From 0d65525b2e69d6f8f37cf3b4e1620281925bebbb Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 26 Apr 2012 15:24:22 +0200
Subject: [PATCH 08/14] Add extension to binaries too

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 36ede3d..be0abfd 100644
--- a/Makefile
+++ b/Makefile
@@ -44,10 +44,10 @@ OBJS= blocksort.o  \
 all: libbz2.a bzip2 bzip2recover
 
 bzip2: libbz2.a bzip2.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2$(EXT) bzip2.o -L. -lbz2
 
 bzip2recover: bzip2recover.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover bzip2recover.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover$(EXT) bzip2recover.o
 
 libbz2.a: $(OBJS)
 	rm -f libbz2.a
-- 
1.8.3.1

