From 88e643e446a88523e1cd5eb333b707a6d36ab7c9 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 18 Oct 2012 16:49:38 +0200
Subject: [PATCH 11/14] create libbz2.dll.a for linking on windows

---
 Makefile-libbz2_dll | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile-libbz2_dll b/Makefile-libbz2_dll
index bd15a84..e021d31 100644
--- a/Makefile-libbz2_dll
+++ b/Makefile-libbz2_dll
@@ -43,13 +43,14 @@ OBJS= blocksort.o  \
       bzlib.o
 
 all: $(OBJS)
-	$(CC) $(BZIP2_LDFLAGS) -shared -o libbz2.dll $(OBJS)
+	$(CC) $(BZIP2_LDFLAGS) -shared -o libbz2.dll -Wl,--out-implib,libbz2.dll.a $(OBJS)
 
 clean: 
 	rm -f $(OBJS) libbz2.dll
 
 install:
 	cp -f libbz2.dll $(PREFIX)/bin
+	cp -f libbz2.dll.a $(PREFIX)/lib
 
 blocksort.o: blocksort.c
 	$(CC) $(BZIP2_CFLAGS) -c blocksort.c
-- 
1.8.3.1

