From f1eb77b885520ad46f7258f57bbb8b4e8418a381 Mon Sep 17 00:00:00 2001
From: Thibault Saunier <thibault.saunier@collabora.com>
Date: Fri, 25 Jan 2013 09:15:17 -0300
Subject: [PATCH 12/14] Do not hardcode CC so it can be cross compiled for iOS

---
 Makefile-libbz2_dylib | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile-libbz2_dylib b/Makefile-libbz2_dylib
index 06f9bd4..9fec33d 100644
--- a/Makefile-libbz2_dylib
+++ b/Makefile-libbz2_dylib
@@ -22,7 +22,9 @@
 
 
 SHELL=/bin/sh
+ifndef CC
 CC=gcc
+endif
 BIGFILES=-D_FILE_OFFSET_BITS=64
 BZIP2_CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
 BZIP2_LDFLAGS=
-- 
1.8.3.1

