From b957312792ec3986a9c31b02a44853a6cdfae478 Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@gnome.org>
Date: Wed, 25 Feb 2015 11:59:56 +0100
Subject: [PATCH] configure: Set gettext to version 0.19

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7cd9542..e3d07b5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -351,7 +351,7 @@ GOBJECT_INTROSPECTION_CHECK([0.9.5])
 
 dnl === Localization ==========================================================
 
-AM_GNU_GETTEXT_VERSION([0.18])
+AM_GNU_GETTEXT_VERSION([0.19])
 AM_GNU_GETTEXT([external])
 
 GETTEXT_PACKAGE="json-glib-1.0"
-- 
2.1.0

