From c39b87713033ae187f9b87ef51209c38b0e32261 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Thu, 11 Oct 2012 14:07:03 +0200
Subject: [PATCH 2/8] Try fixing shared build

---
 Version_script.in | 44 ++++++++++++++++++++++----------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/Version_script.in b/Version_script.in
index 7f22f2f..2b8d834 100644
--- a/Version_script.in
+++ b/Version_script.in
@@ -34,28 +34,28 @@
 		ov_comment;
 		ov_read;
 
-		vorbis_info_init;
-		vorbis_info_clear;
-		vorbis_info_blocksize;
-		vorbis_comment_init;
-		vorbis_comment_add;
-		vorbis_comment_add_tag;
-		vorbis_comment_query;
-		vorbis_comment_query_count;
-		vorbis_comment_clear;
-		vorbis_block_init;
-		vorbis_block_clear;
-		vorbis_dsp_clear;
-		vorbis_synthesis_idheader;
-		vorbis_synthesis_headerin;
-		vorbis_synthesis_init;
-		vorbis_synthesis_restart;
-		vorbis_synthesis;
-		vorbis_synthesis_trackonly;
-		vorbis_synthesis_blockin;
-		vorbis_synthesis_pcmout;
-		vorbis_synthesis_read;
-		vorbis_packet_blocksize;
+		ivorbis_info_init;
+		ivorbis_info_clear;
+		ivorbis_info_blocksize;
+		ivorbis_comment_init;
+		ivorbis_comment_add;
+		ivorbis_comment_add_tag;
+		ivorbis_comment_query;
+		ivorbis_comment_query_count;
+		ivorbis_comment_clear;
+		ivorbis_block_init;
+		ivorbis_block_clear;
+		ivorbis_dsp_clear;
+		ivorbis_synthesis_idheader;
+		ivorbis_synthesis_headerin;
+		ivorbis_synthesis_init;
+		ivorbis_synthesis_restart;
+		ivorbis_synthesis;
+		ivorbis_synthesis_trackonly;
+		ivorbis_synthesis_blockin;
+		ivorbis_synthesis_pcmout;
+		ivorbis_synthesis_read;
+		ivorbis_packet_blocksize;
 
 	local:
 		*;
-- 
2.1.4

