From e3e85b2c74cd615cf8b1b9a4ed7fe30a6d674515 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 15 Jan 2013 16:16:53 +0100
Subject: [PATCH 4/8] Fix build with the Android NDK for x86

---
 misc.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/misc.h b/misc.h
index 1ae4d2e..c855cdd 100644
--- a/misc.h
+++ b/misc.h
@@ -49,9 +49,7 @@ union magic {
   } halves;
   ogg_int64_t whole;
 };
-#endif 
-
-#if BYTE_ORDER==BIG_ENDIAN
+#else
 union magic {
   struct {
     ogg_int32_t hi;
-- 
2.1.4

