From 9f32c6567cd2b9044b63f11bf754f3db6098cadf Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Mon, 7 Oct 2013 15:22:20 +0200
Subject: [PATCH] configure.in: Always use AC_PROG_CC

---
 configure.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/configure.in b/configure.in
index 3ad98ca..8fad378 100644
--- a/configure.in
+++ b/configure.in
@@ -5,9 +5,7 @@ cp $srcdir/configure.sub $srcdir/config.sub
 
 AC_CANONICAL_HOST
 
-if test -z "$CC"; then
-	AC_PROG_CC	
-fi
+AC_PROG_CC	
 AC_PROG_RANLIB
 AC_CHECK_PROG(AR,ar,ar)
 AC_CHECK_PROG(INSTALL,install,install)
-- 
1.8.3.1

