From 0fce59ff8d250a59ccc86894b8f1459070bf6ddb Mon Sep 17 00:00:00 2001
From: Adrian Johnson <ajohnson@redneon.com>
Date: Mon, 2 Oct 2023 15:29:25 +1030
Subject: [PATCH] Fix surface type mismatch error in pdf-interchange

---
 src/cairo-pdf-interchange.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/cairo-pdf-interchange.c b/src/cairo-pdf-interchange.c
index 2f97b081c..9d7aa42fd 100644
--- a/src/cairo-pdf-interchange.c
+++ b/src/cairo-pdf-interchange.c
@@ -52,6 +52,7 @@
 #include "cairo-array-private.h"
 #include "cairo-error-private.h"
 #include "cairo-output-stream-private.h"
+#include "cairo-recording-surface-inline.h"
 #include "cairo-recording-surface-private.h"
 #include "cairo-surface-snapshot-inline.h"
 
@@ -2071,7 +2072,9 @@ _cairo_pdf_interchange_struct_tree_requires_recording_surface (
     if (_cairo_surface_is_snapshot (recording_surface))
 	free_me = recording_surface = _cairo_surface_snapshot_get_target (recording_surface);
 
-    if (_cairo_recording_surface_has_tags (recording_surface)) {
+    if (_cairo_surface_is_recording (recording_surface) &&
+	_cairo_recording_surface_has_tags (recording_surface))
+    {
 	/* Check if tags are to be ignored in this source */
 	switch (source_type) {
 	    case CAIRO_ANALYSIS_SOURCE_PAINT:
-- 
2.43.0

