From edcf449bd33c087ffa40cb05de4b7ed74f3eec61 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 28 Oct 2013 18:19:29 +0100
Subject: [PATCH 1/2] Fix build in Android

---
 configure.ac          | 1 +
 pixman/pixman-utils.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 3654d62..271d02d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -934,6 +934,7 @@ if test $ac_cv_tls = none ; then
     PIXMAN_CHECK_PTHREAD([CFLAGS="-pthread"; LDFLAGS="-pthread"])
     PIXMAN_CHECK_PTHREAD([CFLAGS="-D_REENTRANT"; LIBS="-lpthread"])
     PIXMAN_CHECK_PTHREAD([CFLAGS="-D_REENTRANT"; LDFLAGS="-lroot"])
+    PIXMAN_CHECK_PTHREAD([CFLAGS=""; LDFLAGS="-lc"])
     
     if test $support_for_pthread_setspecific = yes; then
 	CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
diff --git a/pixman/pixman-utils.c b/pixman/pixman-utils.c
index f31171f..9b01e68 100644
--- a/pixman/pixman-utils.c
+++ b/pixman/pixman-utils.c
@@ -27,6 +27,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 
 #include "pixman-private.h"
 
-- 
1.8.1.2

