From 358d3881468c0c75cafa0247b01d28845d45d26e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Wed, 22 Aug 2012 15:10:56 +0200
Subject: [PATCH 2/3] Enable CPU detection on Android

---
 configure.ac       | 5 +++++
 pixman/Makefile.am | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index ba04952..89aad98 100644
--- a/configure.ac
+++ b/configure.ac
@@ -529,8 +529,13 @@ case $host_os in
 	 SSE2_LDFLAGS="$HWCAP_LDFLAGS"
       fi
       ;;
+   linux-android*)
+      is_android="yes"
+      ;;
 esac
 
+AM_CONDITIONAL(IS_ANDROID, test "x$is_android" = "xyes")
+
 AC_SUBST(LS_CFLAGS)
 AC_SUBST(IWMMXT_CFLAGS)
 AC_SUBST(MMX_CFLAGS)
diff --git a/pixman/Makefile.am b/pixman/Makefile.am
index 581b6f6..0e0fbeb 100644
--- a/pixman/Makefile.am
+++ b/pixman/Makefile.am
@@ -114,6 +114,11 @@ libpixman-iwmmxt.la: libpixman_iwmmxt_la-pixman-mmx.lo $(libpixman_iwmmxt_la_DEP
 	$(AM_V_CCLD)$(libpixman_iwmmxt_la_LINK) libpixman_iwmmxt_la-pixman-mmx.lo $(libpixman_iwmmxt_la_LIBADD) $(LIBS)
 endif
 
+if IS_ANDROID
+libpixman_1_la_SOURCES += cpu-features.c
+noinst_HEADERS = cpu-features.h
+endif
+
 # mips dspr2 code
 if USE_MIPS_DSPR2
 noinst_LTLIBRARIES += libpixman-mips-dspr2.la
-- 
2.14.3

