From 70c7fe8163b2d76d7f7e2d68d85b21c30ad45348 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 7 Mar 2012 13:04:25 +0100
Subject: [PATCH 2/6] Update makefiles

---
 configure       | 45 ++++++++++++++++++++++++++++++++++++++++-----
 doc/Makefile.in | 13 +++++++------
 2 files changed, 47 insertions(+), 11 deletions(-)

diff --git a/configure b/configure
index 9703bcb..2864b4e 100755
--- a/configure
+++ b/configure
@@ -833,6 +833,9 @@ HAVE_PDFLATEX
 HAVE_DOXYGEN_FALSE
 HAVE_DOXYGEN_TRUE
 HAVE_DOXYGEN
+BUILD_DOC_FALSE
+BUILD_DOC_TRUE
+BUILD_DOC
 ACLOCAL_AMFLAGS
 OTOOL64
 OTOOL
@@ -961,6 +964,7 @@ enable_libtool_lock
 enable_spec
 enable_valgrind_testing
 enable_asm
+enable_doc
 with_ogg
 with_ogg_libraries
 with_ogg_includes
@@ -1636,6 +1640,7 @@ Optional Features:
                           optimize for fast installation [default=yes]
   --disable-libtool-lock  avoid locking (might break parallel builds)
   --disable-spec          do not build spec
+  --disable-doc           do not build documentation
   --enable-valgrind-testing     enable running of tests inside Valgrind
   --disable-asm           disable assembly optimizations
   --disable-oggtest       Do not try to compile and run a test Ogg program
@@ -11604,6 +11609,33 @@ if test $HAVE_DOXYGEN = "false"; then
 $as_echo "$as_me: WARNING: *** doxygen not found, API documentation will not be built" >&2;}
 fi
 
+ac_build_doc=yes
+# Check whether --enable-doc was given.
+if test "${enable_doc+set}" = set; then
+  enableval=$enable_doc;
+       if test "x$enableval" = "xno"; then
+         ac_build_doc=$enableval
+       fi
+else
+       ac_build_doc=yes
+fi
+
+doc_build="no"
+if test $HAVE_DOXYGEN = "yes"; then
+  if test $ac_build_doc = "yes"; then
+    doc_build="yes"
+  fi
+fi
+
+BUILD_DOC=$doc_build
+if test "x${doc_build}" != xno; then
+  BUILD_DOC_TRUE=
+  BUILD_DOC_FALSE='#'
+else
+  BUILD_DOC_TRUE='#'
+  BUILD_DOC_FALSE=
+fi
+
 BUILD_SPEC="false"
 ac_build_spec=yes
 # Check whether --enable-spec was given.
@@ -14059,6 +14091,13 @@ $as_echo "$as_me: error: conditional \"BUILD_SPEC\" was never defined.
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${BUILD_DOC_TRUE}" && test -z "${BUILD_DOC_FALSE}"; then
+  { { $as_echo "$as_me:$LINENO: error: conditional \"BUILD_DOC\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+$as_echo "$as_me: error: conditional \"BUILD_DOC\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${CPU_x86_64_TRUE}" && test -z "${CPU_x86_64_FALSE}"; then
   { { $as_echo "$as_me:$LINENO: error: conditional \"CPU_x86_64\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
@@ -16419,11 +16458,7 @@ fi
   exec_prefix=$exec_prefix_save
 
 
-if test $HAVE_DOXYGEN = "false"; then
-  doc_build="no"
-else
-  doc_build="yes"
-fi
+
 if test $BUILD_SPEC = "false"; then
   spec_build="no"
 else
diff --git a/doc/Makefile.in b/doc/Makefile.in
index aae84ac..ec7f978 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -90,6 +90,7 @@ GETOPT_OBJS = @GETOPT_OBJS@
 GREP = @GREP@
 HAVE_BIBTEX = @HAVE_BIBTEX@
 HAVE_DOXYGEN = @HAVE_DOXYGEN@
+BUILD_DOC = @BUILD_DOC@
 HAVE_PDFLATEX = @HAVE_PDFLATEX@
 HAVE_PKG_CONFIG = @HAVE_PKG_CONFIG@
 HAVE_TRANSFIG = @HAVE_TRANSFIG@
@@ -451,12 +452,12 @@ uninstall-info: uninstall-info-recursive
 	uninstall-info-recursive uninstall-local uninstall-recursive
 
 
-@HAVE_DOXYGEN_TRUE@doxygen-build.stamp: Doxyfile $(top_srcdir)/include/theora/*.h
-@HAVE_DOXYGEN_TRUE@	doxygen
-@HAVE_DOXYGEN_TRUE@	touch doxygen-build.stamp
-@HAVE_DOXYGEN_FALSE@doxygen-build.stamp:
-@HAVE_DOXYGEN_FALSE@	echo "*** Warning: Doxygen not found; documentation will not be built."
-@HAVE_DOXYGEN_FALSE@	touch doxygen-build.stamp
+@BUILD_DOC_TRUE@doxygen-build.stamp: Doxyfile $(top_srcdir)/include/theora/*.h
+@BUILD_DOC_TRUE@	doxygen
+@BUILD_DOC_TRUE@	touch doxygen-build.stamp
+@BUILD_DOC_FALSE@doxygen-build.stamp:
+@BUILD_DOC_FALSE@	echo "*** Warning: Doxygen not found; documentation will not be built."
+@BUILD_DOC_FALSE@	touch doxygen-build.stamp
 
 dist-hook:
 	if test -d libtheora; then \
-- 
1.8.3.1

