From 1731232552adb44ecf18f3820648fd88e2a3c7db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Wed, 22 Aug 2012 15:58:57 +0200
Subject: [PATCH 2/6] Fix linking of theora encoder library

---
 lib/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index 89ce261..88c4990 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -78,6 +78,7 @@ encoder_uniq_sources = \
 encoder_sources = \
 	apiwrapper.c \
 	fragment.c \
+	info.c \
 	idct.c \
 	internal.c \
 	state.c \
-- 
2.7.4

