From 641cb8dd2db6a816642d747dee161275b2fd1350 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 25 Jan 2013 16:35:36 +0100
Subject: [PATCH 3/6] Use our automake version and not an older one

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index d833491..d888e11 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AUTOMAKE_OPTIONS = foreign 1.6 dist-zip dist-bzip2
+AUTOMAKE_OPTIONS = foreign dist-zip dist-bzip2
 
 if THEORA_ENABLE_EXAMPLES
 EXAMPLES_DIR = examples
-- 
2.7.4

