From 86f7e95f946dafe99cd3fd7e1d19570191516b8c Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Mon, 7 Oct 2013 13:15:20 +0200
Subject: [PATCH 4/6] examples: Don't use png_sizeof

---
 examples/png2theora.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/examples/png2theora.c b/examples/png2theora.c
index 71a0d0f..ed24c2a 100644
--- a/examples/png2theora.c
+++ b/examples/png2theora.c
@@ -462,9 +462,9 @@ png_read(const char *pathname, unsigned int *w, unsigned int *h, unsigned char *
   png_set_strip_alpha(png_ptr);
 
   row_data = (png_bytep)png_malloc(png_ptr,
-    3*height*width*png_sizeof(*row_data));
+    3*height*width*sizeof(*row_data));
   row_pointers = (png_bytep *)png_malloc(png_ptr,
-    height*png_sizeof(*row_pointers));
+    height*sizeof(*row_pointers));
   for(y = 0; y < height; y++) {
     row_pointers[y] = row_data + y*(3*width);
   }
-- 
2.7.4

