From bf1a0a574062f8f119213a12746f4875a4b05c75 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Tue, 6 Nov 2018 14:56:07 +1100
Subject: [PATCH 5/6] build: allow skipping configure in autogen.sh

---
 autogen.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index bbca69d..492b9a7 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -126,4 +126,6 @@ echo "  autoconf"
 autoconf || exit 1
 
 cd $olddir
-$srcdir/configure --enable-maintainer-mode "$@" && echo
+if test "x$NOCONFIGURE" = "x"; then
+  $srcdir/configure --enable-maintainer-mode "$@" && echo
+fi
-- 
2.7.4

