From d04f32348c987210ab5cd2f8ee1fc1385ae7296f Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Wed, 27 May 2020 00:43:26 +0530
Subject: [PATCH 03/22] glib: getpid() is available on Win32, but not WinRT

---
 glib/gmessages.c | 18 ++++++++++++------
 glib/gslice.c    | 11 +++++++++--
 2 files changed, 21 insertions(+), 8 deletions(-)

diff --git a/glib/gmessages.c b/glib/gmessages.c
index bb1ab8f..201192f 100644
--- a/glib/gmessages.c
+++ b/glib/gmessages.c
@@ -203,13 +203,15 @@
 #endif
 
 #ifdef G_OS_WIN32
-#include <process.h>		/* For getpid() */
+#ifndef G_WINAPI_ONLY_APP
+#include <process.h>		/* For getpid(), not available on UWP */
+#endif
 #include <io.h>
 #  include <windows.h>
 
 #ifndef ENABLE_VIRTUAL_TERMINAL_PROCESSING
 #define ENABLE_VIRTUAL_TERMINAL_PROCESSING 0x0004
-#endif
+#endif /* G_OS_WIN32 */
 
 #if defined (_MSC_VER) && (_MSC_VER >=1400)
 /* This is ugly, but we need it for isatty() in case we have bad fd's,
@@ -2253,7 +2255,11 @@ g_log_writer_format_fields (GLogLevelFlags   log_level,
       (log_level & G_LOG_LEVEL_MASK))
     {
       const gchar *prg_name = g_get_prgname ();
+#ifndef G_WINAPI_ONLY_APP
       gulong pid = getpid ();
+#else
+      gulong pid = 0UL;
+#endif
 
       if (prg_name == NULL)
         g_string_append_printf (gstring, "(process:%lu): ", pid);
@@ -2741,7 +2747,7 @@ _g_log_writer_fallback (GLogLevelFlags   log_level,
       write_string_sized (stream, field->value, field->length);
     }
 
-#ifndef G_OS_WIN32
+#ifndef G_WINAPI_ONLY_APP
   {
     gchar pid_string[FORMAT_UNSIGNED_BUFSIZE];
 
@@ -2941,7 +2947,7 @@ _g_log_fallback_handler (const gchar   *log_domain,
 			 gpointer       unused_data)
 {
   gchar level_prefix[STRING_BUFFER_SIZE];
-#ifndef G_OS_WIN32
+#ifndef G_WINAPI_ONLY_APP
   gchar pid_string[FORMAT_UNSIGNED_BUFSIZE];
 #endif
   FILE *stream;
@@ -2957,7 +2963,7 @@ _g_log_fallback_handler (const gchar   *log_domain,
   if (!message)
     message = "(NULL) message";
 
-#ifndef G_OS_WIN32
+#ifndef G_WINAPI_ONLY_APP
   format_unsigned (pid_string, getpid (), 10);
 #endif
 
@@ -2966,7 +2972,7 @@ _g_log_fallback_handler (const gchar   *log_domain,
   else
     write_string (stream, "\n** ");
 
-#ifndef G_OS_WIN32
+#ifndef G_WINAPI_ONLY_APP
   write_string (stream, "(process:");
   write_string (stream, pid_string);
   write_string (stream, "): ");
diff --git a/glib/gslice.c b/glib/gslice.c
index 4c758c3..0b700bb 100644
--- a/glib/gslice.c
+++ b/glib/gslice.c
@@ -29,9 +29,12 @@
 #ifdef G_OS_UNIX
 #include <unistd.h>             /* sysconf() */
 #endif
+
 #ifdef G_OS_WIN32
 #include <windows.h>
-#include <process.h>
+#ifndef G_WINAPI_ONLY_APP
+#include <process.h>            /* getpid(), not available on UWP */
+#endif
 #endif
 
 #include <stdio.h>              /* fputs */
@@ -1467,10 +1470,14 @@ mem_error (const char *format,
 {
   const char *pname;
   va_list args;
+  gulong pid = 0;
   /* at least, put out "MEMORY-ERROR", in case we segfault during the rest of the function */
   fputs ("\n***MEMORY-ERROR***: ", stderr);
   pname = g_get_prgname();
-  g_fprintf (stderr, "%s[%ld]: GSlice: ", pname ? pname : "", (long)getpid());
+#ifndef G_WINAPI_ONLY_APP
+  pid = getpid();
+#endif
+  g_fprintf (stderr, "%s[%lu]: GSlice: ", pname ? pname : "", pid);
   va_start (args, format);
   g_vfprintf (stderr, format, args);
   va_end (args);
-- 
2.27.0.windows.1

