From 4fec54fcce7d67f3d653028ef53a3f581026cc6b Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 30 Dec 2022 02:14:39 +0530
Subject: [PATCH 09/24] gutils: GetModuleHandle is not available on WinRT

---
 glib/gutils.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index d8e89680c..1987eff12 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2512,7 +2512,7 @@ get_module_for_address (gconstpointer address)
   /* Holds the g_utils_global lock */
 
   HMODULE hmodule = NULL;
-
+#ifndef G_WINAPI_ONLY_APP
   if (!address)
     return NULL;
 
@@ -2524,7 +2524,7 @@ get_module_for_address (gconstpointer address)
       VirtualQuery (address, &mbi, sizeof (mbi));
       hmodule = (HMODULE) mbi.AllocationBase;
     }
-
+#endif
   return hmodule;
 }
 
-- 
2.44.0.windows.1

