From 3190f90874cd577ed557b3eed0b703da48a00eab Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Tue, 5 Jul 2016 18:07:45 -0400
Subject: [PATCH 1/4] Protect against unsupported CPU types

---
 configure.ac | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2ca5ea9..d6c2457 100644
--- a/configure.ac
+++ b/configure.ac
@@ -75,7 +75,8 @@ AS_CASE(["${host}"],
         [
          PLATFORM_CFLAGS="-DWEBRTC_WIN -D_WIN32 -U__STRICT_ANSI__"
          HAVE_WIN=1
-        ]
+        ],
+    [AC_MSG_ERROR([Unsupported host $host])]
 )
 AC_SUBST(PLATFORM_CFLAGS)
 AM_CONDITIONAL(HAVE_POSIX, [test "x${HAVE_POSIX}" = "x1"])
@@ -96,8 +97,9 @@ AS_CASE(["${host_cpu}"],
         [
          HAVE_ARM=1
          ARCH_CFLAGS="-DWEBRTC_ARCH_ARM"
-        ]
+        ],
     # FIXME: Add MIPS support, see webrtc/BUILD.gn for defines
+    [AC_MSG_ERROR([Unsupported CPU type $host_cpu])]
 )
 AM_CONDITIONAL(HAVE_X86, [test "x${HAVE_X86}" = "x1"])
 AM_CONDITIONAL(HAVE_ARM, [test "x${HAVE_ARM}" = "x1"])
-- 
2.7.4

