From a0518156887490fb5172b90da870f97f63bc1864 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20Cr=C3=AAte?= <olivier.crete@collabora.com>
Date: Wed, 2 May 2018 14:13:40 -0400
Subject: [PATCH 4/8] Windows: Only define SRT_API in one place

---
 srtcore/srt.h        | 26 +++++++++++++-------------
 srtcore/srt_compat.h | 18 ------------------
 2 files changed, 13 insertions(+), 31 deletions(-)

diff --git a/srtcore/srt.h b/srtcore/srt.h
index 1c6121f..79113cb 100644
--- a/srtcore/srt.h
+++ b/srtcore/srt.h
@@ -37,12 +37,12 @@ written by
 
 #ifdef WIN32
 
-/* Obsolete way to define MINGW */
-#ifndef __MINGW__
-#if defined(__MINGW32__) || defined(__MINGW64__)
-#define __MINGW__ 1
-#endif
-#endif
+      /* Obsolete way to define MINGW */
+      #ifndef __MINGW__
+        #if defined(__MINGW32__) || defined(__MINGW64__)
+          #define __MINGW__ 1
+        #endif
+      #endif
 
       // Explicitly define 32-bit and 64-bit numbers
       typedef __int32 int32_t;
@@ -55,15 +55,15 @@ written by
          typedef __int64 uint64_t;
       #endif
 
-	#ifdef SRT_DYNAMIC
-      #ifdef SRT_EXPORTS
-         #define SRT_API __declspec(dllexport)
+      #ifdef SRT_DYNAMIC
+        #ifdef SRT_EXPORTS
+          #define SRT_API __declspec(dllexport)
+        #elif !defined(__MINGW__)
+          #define SRT_API __declspec(dllimport)
+        #endif
       #else
-         #define SRT_API __declspec(dllimport)
+        #define SRT_API
       #endif
-	#else
-		#define SRT_API
-	#endif
 #else
    #define SRT_API __attribute__ ((visibility("default")))
 #endif
diff --git a/srtcore/srt_compat.h b/srtcore/srt_compat.h
index 6def89d..4d70616 100644
--- a/srtcore/srt_compat.h
+++ b/srtcore/srt_compat.h
@@ -22,24 +22,6 @@ written by
 
 #include "srt.h"
 
-#ifdef WIN32
-   #ifndef __MINGW__
-      #ifdef SRT_DYNAMIC
-         #ifdef SRT_EXPORTS
-            #define SRT_API __declspec(dllexport)
-         #else
-            #define SRT_API __declspec(dllimport)
-         #endif
-      #else
-         #define SRT_API
-      #endif
-   #else
-      #define SRT_API
-   #endif
-#else
-   #define SRT_API __attribute__ ((visibility("default")))
-#endif
-
 #ifdef WIN32
    // https://msdn.microsoft.com/en-us/library/tcxf1dw6.aspx
    // printf() Format for ssize_t
-- 
2.23.0

