From 067c52a405b87953435077584c94c07658c6f53f Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Sun, 28 Oct 2018 02:10:42 +1100
Subject: [PATCH 5/8] Allow building using gnustl

---
 CMakeLists.txt | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fec872c..c31576b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,6 +87,7 @@ option(ENABLE_GETNAMEINFO "In-logs sockaddr-to-string should do rev-dns" OFF)
 option(USE_GNUTLS "Should use gnutls instead of openssl" OFF)
 option(ENABLE_C_DEPS "Extra library dependencies in srt.pc for C language" OFF)
 option(USE_STATIC_LIBSTDCXX "Should use static rather than shared libstdc++" OFF)
+option(USE_GNUSTL "Get c++ library/headers from the gnustl.pc" OFF)
 
 set(TARGET_srt "srt" CACHE STRING "The name for the haisrt library")
 
@@ -204,6 +205,12 @@ endif()
 
 message (STATUS "SSL libraries: ${SSL_LIBRARIES}")
 
+if ( USE_GNUSTL )
+    pkg_check_modules (GNUSTL REQUIRED gnustl)
+    link_directories(${GNUSTL_LIBRARY_DIRS})
+    include_directories(${GNUSTL_INCLUDE_DIRS})
+endif()
+
 # Detect if the compiler is GNU compatable for flags
 set(HAVE_COMPILER_GNU_COMPAT 0)
 foreach (gnid GNU Intel Clang AppleClang)
@@ -458,7 +465,7 @@ if (srt_libspec_shared)
 	set_property(TARGET ${TARGET_srt}_shared PROPERTY OUTPUT_NAME ${TARGET_srt})
     set_target_properties (${TARGET_srt}_shared PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_shared)
-	target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES})
+	target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES} ${GNUSTL_LIBRARIES} ${GNUSTL_LDFLAGS})
 	if (MICROSOFT)
 		target_link_libraries(${TARGET_srt}_shared PRIVATE ws2_32.lib)
 	elseif (MINGW)
@@ -484,7 +491,7 @@ if (srt_libspec_static)
 	endif()
 
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_static)
-	target_link_libraries(${TARGET_srt}_static PRIVATE ${SSL_LIBRARIES})
+	target_link_libraries(${TARGET_srt}_static PRIVATE ${SSL_LIBRARIES} ${GNUSTL_LIBRARIES} ${GNUSTL_LDFLAGS})
 	if (MICROSOFT)
 		target_link_libraries(${TARGET_srt}_static PRIVATE ws2_32.lib)
 	elseif (MINGW)
@@ -506,7 +513,7 @@ endif()
 
 target_include_directories(haicrypt_virtual PRIVATE  ${SSL_INCLUDE_DIRS})
 
-set (SRT_LIBS_PRIVATE ${SSL_LIBRARIES})
+set (SRT_LIBS_PRIVATE ${SSL_LIBRARIES} ${GNUSTL_LIBRARIES} ${GNUSTL_LDFLAGS})
 
 if (MICROSOFT)
 	set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ws2_32.lib)
@@ -665,6 +672,7 @@ macro(srt_make_application name)
 	set_target_properties(${name} PROPERTIES COMPILE_FLAGS "${CFLAGS_CXX_STANDARD} ${EXTRA_stransmit}" ${FORCE_RPATH})
 
 	target_link_libraries(${name} ${srt_link_library})
+	target_link_libraries(${name} PRIVATE ${GNUSTL_LIBRARIES} ${GNUSTL_LDFLAGS})
 endmacro()
 
 macro(srt_add_application name sources)
-- 
2.23.0

