From 71fd327ea76af18957bff414cf7269a602e7119e Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 7 Jan 2019 11:10:30 +0100
Subject: [PATCH 7/7] Fix build with MinGW v6.0.0

---
 common/win/wintime.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/common/win/wintime.h b/common/win/wintime.h
index d763eaa..781bbe5 100644
--- a/common/win/wintime.h
+++ b/common/win/wintime.h
@@ -3,6 +3,11 @@
 
 #include <winsock2.h>
 #include <windows.h>
+// HACK: This include is a workaround for a bug in the MinGW headers
+// where pthread.h, which defines _POSIX_THREAD_SAFE_FUNCTIONS,
+// has to be included before time.h so that time.h defines
+// localtime_r correctly
+#include <pthread.h>
 #include <time.h>
 
 #ifdef __cplusplus
-- 
2.7.4

