From 420f86d670c845924a8dafe88595abafd7bf67c2 Mon Sep 17 00:00:00 2001
From: Justin Kim <justin.kim@collabora.com>
Date: Tue, 2 Jan 2018 18:22:00 +0900
Subject: [PATCH 8/9] build: Fix for MinGW support

Signed-off-by: Justin Kim <justin.kim@collabora.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bf4fe9..7c95096 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -235,7 +235,7 @@ else()
 endif()
 
 else()
-set (PTHREAD_LIBRARY -lpthreadGC2)
+find_library(PTHREAD_LIBRARY NAMES pthread pthreadGC2 pthreadGC)
 endif() # if (NOT MINGW)
 
 # This is required in some projects that add some other sources
@@ -371,7 +371,7 @@ target_include_directories(${TARGET_haicrypt}
 set_target_properties (${TARGET_haicrypt} PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
 target_link_libraries(${TARGET_haicrypt} PRIVATE ${SRT_LIBS_PRIVATE} ${SSL_LIBRARIES})
 
-if ( WIN32 AND (NOT MINGW AND NOT CYGWIN) )
+if ( WIN32 AND NOT CYGWIN )
 	target_link_libraries(${TARGET_haicrypt} PRIVATE ws2_32.lib)
 	set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ws2_32.lib)
 endif()
-- 
2.15.1

