From e8ee43bd369ea451604f7b8be89918d489a0185b Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Thu, 30 Oct 2014 21:09:41 +0000
Subject: Add ability to build static GST plugins

Note that this only works for 1.0 plugin.
---
diff --git a/configure.ac b/configure.ac
index 5021a1f..744d407 100644
--- a/configure.ac
+++ b/configure.ac
@@ -286,6 +286,24 @@ AS_IF([test "${enable_coverage}" != "no"], [
 ])
 AC_SUBST(CCACHE_DISABLE)
 
+dnl build static plugins or not
+AC_MSG_CHECKING([whether to build static plugins or not])
+AC_ARG_ENABLE(
+  static-plugins,
+  AC_HELP_STRING(
+    [--enable-static-plugins],
+    [build static plugins @<:@default=no@:>@]),
+  [AS_CASE(
+    [$enableval], [no], [], [yes], [],
+    [AC_MSG_ERROR([bad value "$enableval" for --enable-static-plugins])])],
+  [enable_static_plugins=no])
+AC_MSG_RESULT([$enable_static_plugins])
+if test "x$enable_static_plugins" = xyes; then
+  AC_DEFINE(GST_PLUGIN_BUILD_STATIC, 1,
+    [Define if static plugins should be built])
+fi
+AM_CONDITIONAL(GST_PLUGIN_BUILD_STATIC, test "x$enable_static_plugins" = "xyes")
+
 case $host_os in
   solaris*)
     LDFLAGS="$LDFLAGS -lsocket -lnls"
--
cgit v0.9.0.3
