From 4c62d65f6853e9d9650ab1d9bedc7c83d6e23d45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Wed, 28 Dec 2022 16:56:16 +0100
Subject: [PATCH 3/3] meson: define SUPPORT_PCRE2_16 etc as well if those are
 built

---
 meson.build | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 9bcb557..1aff62e 100644
--- a/meson.build
+++ b/meson.build
@@ -93,13 +93,21 @@ foreach f : check_funs
   endif
 endforeach
 
-# TODO: also define SUPPORT_PCRE2_16 and SUPPORT_PCRE2_32 if available?
 config_h_defs += [
   '-DSTDC_HEADERS',
-  '-DSUPPORT_PCRE2_8',
   '-DSUPPORT_UNICODE',
 ]
 
+if not get_option('pcre2-8').disabled()
+  config_h_defs += ['-DSUPPORT_PCRE2_8']
+endif
+if get_option('pcre2-16').enabled()
+  config_h_defs += ['-DSUPPORT_PCRE2_16']
+endif
+if get_option('pcre2-32').enabled()
+  config_h_defs += ['-DSUPPORT_PCRE2_32']
+endif
+
 if get_option('default_library') == 'static'
   static_defs = ['-DPCRE2_STATIC']
   config_h_defs += static_defs
-- 
2.38.1

