From 9bdbe1d8262bc19457a2c43bc78279f2b0be4b47 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 28 Oct 2013 17:50:25 +0100
Subject: [PATCH] win32: fix dll name

---
 win32/Makefile.gcc | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/win32/Makefile.gcc b/win32/Makefile.gcc
index 305be50..3970c79 100644
--- a/win32/Makefile.gcc
+++ b/win32/Makefile.gcc
@@ -30,18 +30,17 @@
 # the DLL name should be changed from "zlib1.dll".
 
 STATICLIB = libz.a
-SHAREDLIB = zlib1.dll
+SHAREDLIB = libz.dll
 IMPLIB    = libz.dll.a
 
 #
 # Set to 1 if shared object needs to be installed
 #
-SHARED_MODE=0
+SHARED_MODE=1
 
 #LOC = -DASMV
 #LOC = -DZLIB_DEBUG -g
 
-PREFIX =
 CC = $(PREFIX)gcc
 CFLAGS = $(LOC) -O3 -Wall
 
-- 
2.11.0

