From 86118d1e85259e82d67249e92b877578fe34e487 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Tue, 13 Feb 2018 17:14:39 +1100
Subject: [PATCH] Fix test builds to use the built libz headers/library

---
 Makefile.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 5a77949..bb1795d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -26,7 +26,7 @@ CFLAGS=-O
 
 SFLAGS=-O
 LDFLAGS=
-TEST_LDFLAGS=-L. libz.a
+TEST_LDFLAGS=-L. libz.a $(LDFLAGS)
 LDSHARED=$(CC)
 CPP=$(CC) -E
 
@@ -144,16 +144,16 @@ match.lo: match.S
 	rm -f _match.s
 
 example.o: $(SRCDIR)test/example.c $(SRCDIR)zlib.h zconf.h
-	$(CC) $(CFLAGS) $(ZINCOUT) -c -o $@ $(SRCDIR)test/example.c
+	$(CC) $(ZINCOUT) $(CFLAGS) -c -o $@ $(SRCDIR)test/example.c
 
 minigzip.o: $(SRCDIR)test/minigzip.c $(SRCDIR)zlib.h zconf.h
-	$(CC) $(CFLAGS) $(ZINCOUT) -c -o $@ $(SRCDIR)test/minigzip.c
+	$(CC) $(ZINCOUT) $(CFLAGS) -c -o $@ $(SRCDIR)test/minigzip.c
 
 example64.o: $(SRCDIR)test/example.c $(SRCDIR)zlib.h zconf.h
-	$(CC) $(CFLAGS) $(ZINCOUT) -D_FILE_OFFSET_BITS=64 -c -o $@ $(SRCDIR)test/example.c
+	$(CC) $(ZINCOUT) $(CFLAGS) -D_FILE_OFFSET_BITS=64 -c -o $@ $(SRCDIR)test/example.c
 
 minigzip64.o: $(SRCDIR)test/minigzip.c $(SRCDIR)zlib.h zconf.h
-	$(CC) $(CFLAGS) $(ZINCOUT) -D_FILE_OFFSET_BITS=64 -c -o $@ $(SRCDIR)test/minigzip.c
+	$(CC) $(ZINCOUT) $(CFLAGS) -D_FILE_OFFSET_BITS=64 -c -o $@ $(SRCDIR)test/minigzip.c
 
 
 adler32.o: $(SRCDIR)adler32.c
-- 
2.16.1

