From 60ec5d91ba521a2379c6c1803f2b587d0eafedb5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <slomo@circular-chaos.org>
Date: Sun, 22 Sep 2013 17:22:23 +0200
Subject: [PATCH 8/9] Fix compilation with clang

---
 source/SoundTouch/cpu_detect_x86.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/SoundTouch/cpu_detect_x86.cpp b/source/SoundTouch/cpu_detect_x86.cpp
index 5ef0246..2c6fb52 100644
--- a/source/SoundTouch/cpu_detect_x86.cpp
+++ b/source/SoundTouch/cpu_detect_x86.cpp
@@ -45,7 +45,7 @@
 
 #if defined(SOUNDTOUCH_ALLOW_X86_OPTIMIZATIONS)
 
-   #if defined(__GNUC__) && defined(__i386__)
+   #if defined(__GNUC__) && defined(__i386__) && !defined(__clang__)
        // gcc
        #include "cpuid.h"
    #elif defined(_M_IX86)
@@ -89,7 +89,7 @@ uint detectCPUextensions(void)
 
 /// If building for a 32bit system and the user wants optimizations.
 /// Keep the _dwDisabledISA test (2 more operations, could be eliminated).
-#elif ((defined(__GNUC__) && defined(__i386__)) \
+#elif ((defined(__GNUC__) && defined(__i386__) && !defined(__clang__)) \
     || defined(_M_IX86))  \
     && defined(SOUNDTOUCH_ALLOW_X86_OPTIMIZATIONS)
 
-- 
2.1.4

