From 0bfeb3d7bec127ccd76e6fad937702f5aef12500 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 29 Oct 2013 17:59:29 +0100
Subject: [PATCH] ios: fix build using .word for assembly constants

---
 arm/ecc-521-modp.asm |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arm/ecc-521-modp.asm b/arm/ecc-521-modp.asm
index fe30580..287cac4 100644
--- a/arm/ecc-521-modp.asm
+++ b/arm/ecc-521-modp.asm
@@ -35,7 +35,11 @@ define(<N>, <lr>)
 	C ecc_521_modp (const struct ecc_curve *ecc, mp_limb_t *rp)
 	.text
 .Lc511:
+#ifdef __APPLE__
+	.word 511
+#else
 	.int 511
+#endif
 
 	.align 2
 
-- 
1.7.4.4

