From ff87370625a68b73d0e1351bc735ef510c26d051 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 9 Mar 2012 18:18:48 +0100
Subject: [PATCH 1/5] Enable windows cross-compilation

---
 configure.ac       | 21 ++++++++++++++++++++-
 python/Makefile.am |  1 +
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 14ac0a8..c3985f2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -811,6 +811,21 @@ dnl
 dnl check for python
 dnl
 
+case "$host" in
+  *-*-mingw*|*-*-cygwin*)
+    platform_win32=yes
+    ;;
+  *)
+    platform_win32=no
+    ;;
+esac
+
+if test "x$platform_win32" = "xyes" ; then
+AM_CONDITIONAL(WITH_PYTHON, test "$PYTHON_INCLUDES" != "")
+pythondir=$am_cv_python_pythondir
+PYTHON_SUBDIR=python
+PYTHON=`which python`
+else
 PYTHON_VERSION=
 PYTHON_INCLUDES=
 PYTHON_SITE_PACKAGES=
@@ -905,6 +920,7 @@ then
 else
     PYTHON_SUBDIR=
 fi
+fi
 AC_SUBST(pythondir)
 AC_SUBST(PYTHON_SUBDIR)
 AC_SUBST(PYTHON_LIBS)
@@ -1574,6 +1590,7 @@ WIN32_EXTRA_LDFLAGS=
 CYGWIN_EXTRA_LDFLAGS=
 CYGWIN_EXTRA_PYTHON_LIBADD=
 WIN32_EXTRA_PYTHON_LIBADD=
+WIN32_EXTRA_PYTHON_CFLAGS=
 case "$host" in
  *-*-mingw*)
  CPPFLAGS="$CPPFLAGS -DWIN32"
@@ -1582,7 +1599,8 @@ case "$host" in
  AC_DEFINE([_WINSOCKAPI_],1,[Using the Win32 Socket implementation])
  if test "${PYTHON}" != ""
  then
-   WIN32_EXTRA_PYTHON_LIBADD="-L${pythondir}/../../libs -lpython$(echo ${PYTHON_VERSION} | tr -d .)"
+   WIN32_EXTRA_PYTHON_LIBADD="${PYTHON_LIBS}"
+   WIN32_EXTRA_PYTHON_CFLAGS="${PYTHON_INCLUDES}"
  fi
  ;;
  *-*-cygwin*)
@@ -1596,6 +1614,7 @@ esac
 AC_SUBST(WIN32_EXTRA_LIBADD)
 AC_SUBST(WIN32_EXTRA_LDFLAGS)
 AC_SUBST(WIN32_EXTRA_PYTHON_LIBADD)
+AC_SUBST(WIN32_EXTRA_PYTHON_CFLAGS)
 AC_SUBST(CYGWIN_EXTRA_LDFLAGS)
 AC_SUBST(CYGWIN_EXTRA_PYTHON_LIBADD)
 
diff --git a/python/Makefile.am b/python/Makefile.am
index 34aed96..8744cc7 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -24,6 +24,7 @@ AM_CPPFLAGS = \
 python_LTLIBRARIES = libxml2mod.la
 
 libxml2mod_la_SOURCES = libxml.c libxml_wrap.h libxml2-py.h libxml2-py.c types.c
+libxml2mod_la_CFLAGS = $(WIN32_EXTRA_PYTHON_CFLAGS)
 libxml2mod_la_LDFLAGS = $(CYGWIN_EXTRA_LDFLAGS) $(WIN32_EXTRA_LDFLAGS) -module -avoid-version \
         $(top_builddir)/libxml2.la $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) $(PYTHON_LIBS)
 
-- 
2.1.0

