From a1c3a1aa10e6477fbe97a5b1bf4c63c13c8d7901 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Tue, 13 Aug 2024 17:28:13 +0000
Subject: [PATCH] meson: Make tests optional

---
 meson.build       | 6 ++++--
 meson_options.txt | 6 ++++++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 988d7ec..5cea50b 100644
--- a/meson.build
+++ b/meson.build
@@ -542,7 +542,8 @@ checks = {
     'testrecurse': [],
 }
 
-foreach check, deps : checks
+if get_option('tests')
+  foreach check, deps : checks
     exe = executable(
         check,
         files(check + '.c'),
@@ -552,7 +553,8 @@ foreach check, deps : checks
     if check != 'testlimits'
         test(check, exe, timeout: 0, workdir: meson.current_source_dir())
     endif
-endforeach
+  endforeach
+endif
 
 sh = find_program('sh', required: false)
 
diff --git a/meson_options.txt b/meson_options.txt
index 0bfbebb..e8feb0a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -204,3 +204,9 @@ option('minimum',
   value: false,
   description: 'build a minimally sized library'
 )
+
+option('tests',
+  type: 'boolean',
+  value: false,
+  description: 'build tests'
+)
-- 
2.49.0

