From 5a75153014bd1c0669415bda1ee899c0ae9c3aad Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 28 Oct 2013 18:07:07 +0100
Subject: [PATCH 3/5] configure: fix python dir prefix

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index c3985f2..e5dbc15 100644
--- a/configure.ac
+++ b/configure.ac
@@ -908,6 +908,7 @@ if test "$with_python" != "no" ; then
 	    fi
 	fi
     fi
+    PYTHON_SITE_PACKAGES='$(prefix)/lib/python$(PYTHON_VERSION)/site-packages'
     pythondir='$(PYTHON_SITE_PACKAGES)'
     PYTHON_LIBS=`python$PYTHON_VERSION-config --ldflags`
 else
-- 
2.1.0

