From 732ef56b0b1315f2cfa7ae8fe1181309ce95e87d Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 28 Jan 2014 11:44:22 +0100
Subject: [PATCH 4/5] Fix build without lzma

---
 xmlIO.c | 6 +++---
 xzlib.c | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/xmlIO.c b/xmlIO.c
index e628ab0..a1a927e 100644
--- a/xmlIO.c
+++ b/xmlIO.c
@@ -35,7 +35,7 @@
 #ifdef HAVE_ZLIB_H
 #include <zlib.h>
 #endif
-#ifdef HAVE_LZMA_H
+#ifdef HAVE_LIBLZMA
 #include <lzma.h>
 #endif
 
@@ -1334,7 +1334,7 @@ xmlGzfileClose (void * context) {
 }
 #endif /* HAVE_ZLIB_H */
 
-#ifdef HAVE_LZMA_H
+#ifdef HAVE_LIBLZMA
 /************************************************************************
  *									*
  *		I/O for compressed file accesses			*
@@ -2328,7 +2328,7 @@ xmlRegisterDefaultInputCallbacks(void) {
     xmlRegisterInputCallbacks(xmlGzfileMatch, xmlGzfileOpen,
 	                      xmlGzfileRead, xmlGzfileClose);
 #endif /* HAVE_ZLIB_H */
-#ifdef HAVE_LZMA_H
+#ifdef HAVE_LIBLZMA
     xmlRegisterInputCallbacks(xmlXzfileMatch, xmlXzfileOpen,
 	                      xmlXzfileRead, xmlXzfileClose);
 #endif /* HAVE_ZLIB_H */
diff --git a/xzlib.c b/xzlib.c
index 0dcb9f4..1cf0c57 100644
--- a/xzlib.c
+++ b/xzlib.c
@@ -8,7 +8,7 @@
  */
 #define IN_LIBXML
 #include "libxml.h"
-#ifdef HAVE_LZMA_H
+#ifdef HAVE_LIBLZMA
 
 #include <string.h>
 #ifdef HAVE_ERRNO_H
@@ -795,4 +795,4 @@ __libxml2_xzclose(xzFile file)
     xmlFree(state);
     return ret ? ret : LZMA_OK;
 }
-#endif /* HAVE_LZMA_H */
+#endif /* HAVE_LIBLZMA */
-- 
2.1.0

