From 2676c2425e918d0c25e2ebf1ddf7980dc0d43d3c Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Fri, 13 Feb 2015 13:05:05 +0100
Subject: [PATCH 5/5] configure.ac: Don't overwrite LDFLAGS

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e5dbc15..080134f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -392,7 +392,7 @@ if test "$with_zlib" = "no"; then
 else
     AC_CHECK_HEADERS(zlib.h,
         [SAVE_LDFLAGS="${LDFLAGS}"
-	 LDFLAGS="-L${Z_DIR}/lib"
+	 LDFLAGS="${LDFLAGS} -L${Z_DIR}/lib"
 	AC_CHECK_LIB(z, gzread,[
 	    AC_DEFINE([HAVE_LIBZ], [1], [Have compression library])
 	    WITH_ZLIB=1
-- 
2.1.0

