From 66d0d569e1aacf1215ecbe1a7c133b4bd7a0802a Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sun, 26 Feb 2012 02:24:59 +0100
Subject: [PATCH 1/4] Honour --with-pic option properly

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 73f2cf7..fccbdd8 100755
--- a/configure
+++ b/configure
@@ -9640,7 +9640,7 @@ _ACEOF
 
 
 
-LIBA52_CFLAGS="$LIBA52_CFLAGS -prefer-non-pic"
+LIBA52_CFLAGS="$LIBA52_CFLAGS $pic_flag"
 
 # Check whether --enable-double or --disable-double was given.
 if test "${enable_double+set}" = set; then
-- 
2.1.4

