From 57d6b23dabc50f2cb62a8f59ad57353925f0a967 Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@centricular.com>
Date: Fri, 20 Feb 2015 16:38:58 +0530
Subject: [PATCH 1/9] build: Fix the min version flag for iOS simulator builds

---
 build/make/configure.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 683b430..cf1dd6a 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -886,8 +886,8 @@ process_common_toolchain() {
       add_ldflags "-mmacosx-version-min=10.12"
       ;;
     *-iphonesimulator-*)
-      add_cflags  "-miphoneos-version-min=${IOS_VERSION_MIN}"
-      add_ldflags "-miphoneos-version-min=${IOS_VERSION_MIN}"
+      add_cflags  "-mios-simulator-version-min=${IOS_VERSION_MIN}"
+      add_ldflags "-mios-simulator-version-min=${IOS_VERSION_MIN}"
       iossim_sdk_dir="$(show_darwin_sdk_path iphonesimulator)"
       if [ -d "${iossim_sdk_dir}" ]; then
         add_cflags  "-isysroot ${iossim_sdk_dir}"
-- 
2.20.1.windows.1

