From aa921dc67b86e743497fa207ff68dd2ed83dad3d Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@centricular.com>
Date: Fri, 20 Feb 2015 16:38:58 +0530
Subject: [PATCH] build: Fix the min version flag for iOS simulator builds

---
 build/make/configure.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index f980180..56e4d0d 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -776,8 +776,8 @@ process_common_toolchain() {
       add_ldflags "-mmacosx-version-min=10.9"
       ;;
     *-iphonesimulator-*)
-      add_cflags  "-miphoneos-version-min=${IOS_VERSION_MIN}"
-      add_ldflags "-miphoneos-version-min=${IOS_VERSION_MIN}"
+      add_cflags  "-mios-simulator-version-min=${IOS_VERSION_MIN}"
+      add_ldflags "-mios-simulator-version-min=${IOS_VERSION_MIN}"
       osx_sdk_dir="$(xcrun --sdk iphonesimulator --show-sdk-path)"
       add_cflags  "-isysroot ${osx_sdk_dir}"
       add_ldflags "-isysroot ${osx_sdk_dir}"
-- 
2.1.0

