From 4becc7c54ad3e7f8e5c9d1d59ba401cd4a2c9b3c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Tue, 9 Oct 2012 13:48:12 +0200
Subject: [PATCH 2/6] Include Android cpu-features.c instead of .h

---
 vpx_ports/arm_cpudetect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vpx_ports/arm_cpudetect.c b/vpx_ports/arm_cpudetect.c
index 8a4b8af..60401f3 100644
--- a/vpx_ports/arm_cpudetect.c
+++ b/vpx_ports/arm_cpudetect.c
@@ -101,7 +101,7 @@ return flags & mask;
 }
 
 #elif defined(__ANDROID__) /* end _MSC_VER */
-#include <cpu-features.h>
+#include <cpu-features.c>
 
 int arm_cpu_caps(void) {
   int flags;
-- 
2.9.3

