From 595891d8d5357aead7d6342cfe6cfbffd046cee4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <slomo@circular-chaos.org>
Date: Sat, 21 Sep 2013 18:03:55 +0200
Subject: [PATCH 5/5] Add support for iOS 7.0 toolchain

---
 build/make/configure.sh | 20 ++++++++++++++------
 configure               |  3 +++
 2 files changed, 17 insertions(+), 6 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 3bd5618..294efce 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -664,6 +664,9 @@
             *ios6*)
                 tgt_os=ios6
                 ;;
+            *ios7*)
+                tgt_os=ios7
+                ;;
             *mingw32*|*cygwin*)
                 [ -z "$tgt_isa" ] && tgt_isa=x86
                 tgt_os=win32
@@ -752,7 +755,7 @@
                 SDK_PATH=${sdk_path}
             fi
             IOS_SDK_ROOT="${SDK_PATH}/SDKs"
-            IOS_SDK_VERSIONS="6.0 6.1"
+            IOS_SDK_VERSIONS="6.0 6.1 7.0"
             for v in ${IOS_SDK_VERSIONS}; do
                 if [ -d "${IOS_SDK_ROOT}/${IOS_PLATFORM}${v}.sdk" ]; then
                     osx_sdk_dir="${IOS_SDK_ROOT}/${IOS_PLATFORM}${v}.sdk"
@@ -790,9 +793,9 @@
             add_cflags  "-mmacosx-version-min=10.9"
             add_ldflags "-mmacosx-version-min=10.9"
             ;;
-        *-ios6-*)
-            add_cflags  "-miphoneos-version-min=6.0"
-            add_ldflags "-miphoneos-version-min=6.0"
+        *-ios*)
+            add_cflags  "-miphoneos-version-min=5.1"
+            add_ldflags "-miphoneos-version-min=5.1"
             ;;
     esac
 
@@ -989,9 +992,14 @@
          ;;
         ios*)
             TOOLCHAIN_PATH=${SDK_PATH}/usr/bin
-            CC=${TOOLCHAIN_PATH}/llvm-gcc-4.2
+            if test "x$tgt_os" = "xios7"; then
+                CC=clang
+                LD=clang
+            else
+                CC=${TOOLCHAIN_PATH}/llvm-gcc-4.2
+                LD=${TOOLCHAIN_PATH}/llvm-gcc-4.2
+            fi
             AR=${TOOLCHAIN_PATH}/ar
-            LD=${TOOLCHAIN_PATH}/llvm-gcc-4.2
             AS=${TOOLCHAIN_PATH}/as
             STRIP=${TOOLCHAIN_PATH}/strip
             NM=${TOOLCHAIN_PATH}/nm
diff --git a/configure b/configure
index d219584..cdf4a69 100755
--- a/configure
+++ b/configure
@@ -95,12 +95,14 @@
 all_platforms="${all_platforms} armv6-linux-gcc"
 all_platforms="${all_platforms} armv6-none-rvct"
 all_platforms="${all_platforms} armv6-ios6-gcc"
+all_platforms="${all_platforms} armv6-ios7-gcc"
 all_platforms="${all_platforms} armv7-android-gcc"   #neon Cortex-A8
 all_platforms="${all_platforms} armv7-darwin-gcc"    #neon Cortex-A8
 all_platforms="${all_platforms} armv7-linux-rvct"    #neon Cortex-A8
 all_platforms="${all_platforms} armv7-linux-gcc"     #neon Cortex-A8
 all_platforms="${all_platforms} armv7-none-rvct"     #neon Cortex-A8
 all_platforms="${all_platforms} armv7-ios6-gcc"      #neon Cortex-A8
+all_platforms="${all_platforms} armv7-ios7-gcc"      #neon Cortex-A8
 all_platforms="${all_platforms} armv7-win32-vs11"
 all_platforms="${all_platforms} armv7-win32-vs12"
 all_platforms="${all_platforms} mips32-linux-gcc"
@@ -121,6 +123,7 @@
 all_platforms="${all_platforms} x86-darwin12-gcc"
 all_platforms="${all_platforms} x86-darwin13-gcc"
 all_platforms="${all_platforms} x86-ios6-gcc"
+all_platforms="${all_platforms} x86-ios7-gcc"
 all_platforms="${all_platforms} x86-linux-gcc"
 all_platforms="${all_platforms} x86-linux-icc"
 all_platforms="${all_platforms} x86-os2-gcc"
-- 
1.8.3.1

