From 82debca13c31da4fced144e829bdaa102db79674 Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Sat, 14 Feb 2015 15:09:40 +0100
Subject: [PATCH 5/6] configure: Add Android ARM64 support

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index ea8e015..153f095 100755
--- a/configure
+++ b/configure
@@ -97,6 +97,7 @@ EOF
 
 # all_platforms is a list of all supported target platforms. Maintain
 # alphabetically by architecture, generic-gnu last.
+all_platforms="${all_platforms} arm64-android-gcc"
 all_platforms="${all_platforms} arm64-darwin-gcc"
 all_platforms="${all_platforms} arm64-linux-gcc"
 all_platforms="${all_platforms} armv6-linux-rvct"
-- 
2.9.3

