From f104b6a2de3da66aab7f1771d0cc49ff9ac9c5b8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Mon, 28 Dec 2015 17:20:41 +0200
Subject: [PATCH 6/6] Don't embed bitcode on iOS until we support that properly

---
 build/make/configure.sh | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index e56108e..528174b 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1078,11 +1078,6 @@ EOF
 
           asm_conversion_cmd="${source_path}/build/make/ads2gas_apple.pl"
 
-          if [ "$(show_darwin_sdk_major_version iphoneos)" -gt 8 ]; then
-            check_add_cflags -fembed-bitcode
-            check_add_asflags -fembed-bitcode
-            check_add_ldflags -fembed-bitcode
-          fi
           ;;
 
         linux*)
-- 
2.9.3

