From 94d9c8f3079f298f0fd959b0672c491f7e2e57db Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward@collabora.com>
Date: Wed, 16 Apr 2014 13:57:11 +0200
Subject: [PATCH] Add Armv7s ios support

---
 build/make/configure.sh | 5 ++++-
 configure               | 1 +
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 294efce..ff47a31 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -604,6 +604,9 @@
             armv6*)
                 tgt_isa=armv6
                 ;;
+            armv7s*)
+                tgt_isa=armv7s
+                ;;
             armv7*-hardfloat*)
                 tgt_isa=armv7
                 float_abi=hard
@@ -815,7 +818,7 @@
     arm*)
         # on arm, isa versions are supersets
         case ${tgt_isa} in
-        armv7)
+        armv7|armv7s)
             soft_enable neon
             soft_enable media
             soft_enable edsp
diff --git a/configure b/configure
index cdf4a69..0f24b93 100755
--- a/configure
+++ b/configure
@@ -103,6 +103,7 @@
 all_platforms="${all_platforms} armv7-none-rvct"     #neon Cortex-A8
 all_platforms="${all_platforms} armv7-ios6-gcc"      #neon Cortex-A8
 all_platforms="${all_platforms} armv7-ios7-gcc"      #neon Cortex-A8
+all_platforms="${all_platforms} armv7s-ios7-gcc"     #neon Cortex-A9
 all_platforms="${all_platforms} armv7-win32-vs11"
 all_platforms="${all_platforms} armv7-win32-vs12"
 all_platforms="${all_platforms} mips32-linux-gcc"
