From 98ae18374a0b54c69b6fdc9b55fb76cc3236691d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Mon, 22 Sep 2014 16:01:57 +0200
Subject: [PATCH] Add support for iOS simulator on x86-64

---
 build/make/configure.sh | 12 ++++++++++--
 configure               |  1 +
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 44b443a..9076104 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -797,8 +797,16 @@ process_common_toolchain() {
             add_ldflags "-mmacosx-version-min=10.9"
             ;;
         *-ios*)
-            add_cflags  "-miphoneos-version-min=5.1"
-            add_ldflags "-miphoneos-version-min=5.1"
+            case "$tgt_isa" in
+                arm*)
+                    add_cflags  "-miphoneos-version-min=5.1"
+                    add_ldflags "-miphoneos-version-min=5.1"
+                    ;;
+                x86*)
+                    add_cflags  "-mios-simulator-version-min=5.1"
+                    add_ldflags "-mios-simulator-version-min=5.1"
+                    ;;
+                esac
             ;;
     esac
 
diff --git a/configure b/configure
index f2423a2..833abb4 100755
--- a/configure
+++ b/configure
@@ -141,6 +141,7 @@ all_platforms="${all_platforms} x86_64-darwin10-gcc"
 all_platforms="${all_platforms} x86_64-darwin11-gcc"
 all_platforms="${all_platforms} x86_64-darwin12-gcc"
 all_platforms="${all_platforms} x86_64-darwin13-gcc"
+all_platforms="${all_platforms} x86_64-ios7-gcc"
 all_platforms="${all_platforms} x86_64-linux-gcc"
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
-- 
1.9.3 (Apple Git-50)

