From c3e3ae1d5ecd3957a68c96fcbb940220405cf1f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Mon, 27 Oct 2014 19:25:18 +0100
Subject: [PATCH] Add ARM64 iOS support

---
 build/make/configure.sh | 7 ++++++-
 configure               | 1 +
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 9076104..47c2402 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -618,6 +618,9 @@ process_common_toolchain() {
             armv5te*)
                 tgt_isa=armv5te
                 ;;
+            arm64*)
+                tgt_isa=arm64
+                ;;
             *x86_64*|*amd64*)
                 tgt_isa=x86_64
                 ;;
@@ -826,6 +829,8 @@ process_common_toolchain() {
     arm*)
         # on arm, isa versions are supersets
         case ${tgt_isa} in
+        arm64)
+            ;;
         armv7|armv7s)
             soft_enable neon
             soft_enable media
@@ -874,7 +879,7 @@ EOF
                 if [ -z "${tune_cpu}" ]; then
                     tune_cpu=cortex-a8
                 fi
-            else
+            elif [ ${tgt_isa} != "arm64" ]; then
                 check_add_cflags -march=${tgt_isa}
                 check_add_asflags -march=${tgt_isa}
             fi
diff --git a/configure b/configure
index 833abb4..078377e 100755
--- a/configure
+++ b/configure
@@ -104,6 +104,7 @@ all_platforms="${all_platforms} armv7-none-rvct"     #neon Cortex-A8
 all_platforms="${all_platforms} armv7-ios6-gcc"      #neon Cortex-A8
 all_platforms="${all_platforms} armv7-ios7-gcc"      #neon Cortex-A8
 all_platforms="${all_platforms} armv7s-ios7-gcc"     #neon Cortex-A9
+all_platforms="${all_platforms} arm64-ios7-gcc"      #neon Cortex-A9
 all_platforms="${all_platforms} armv7-win32-vs11"
 all_platforms="${all_platforms} armv7-win32-vs12"
 all_platforms="${all_platforms} mips32-linux-gcc"
-- 
diff --git a/third_party/libyuv/source/scale.c b/third_party/libyuv/source/scale.c
index 3c30b55..e2a078e 100644
--- a/third_party/libyuv/source/scale.c
+++ b/third_party/libyuv/source/scale.c
@@ -58,6 +58,10 @@ void SetUseReferenceImpl(int use) {
  *
  */
 
+#if defined(__arm64__) && defined(__ARM_NEON__)
+#undef __ARM_NEON__
+#endif
+
 #if defined(__ARM_NEON__) && !defined(YUV_DISABLE_ASM)
 #define HAS_SCALEROWDOWN2_NEON
 void ScaleRowDown2_NEON(const uint8* src_ptr, int  src_stride,

1.9.3 (Apple Git-50)

