From 72c9cf274021a795ee54a8da308eb249126e3b5f Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 29 Feb 2012 18:12:14 +0100
Subject: [PATCH 1/2] Fix linking in windows compilation

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 89bfd9d..555df61 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -36,7 +36,7 @@ libdvdnav_la_SOURCES = \
 
 libdvdnav_la_LDFLAGS = \
 	-version-info $(DVDNAV_LT_CURRENT):$(DVDNAV_LT_REVISION):$(DVDNAV_LT_AGE) \
-	-export-symbols-regex "^dvdnav"
+	-export-symbols-regex "^dvdnav" $(DVDREAD_LIBS)
 libdvdnav_la_LIBADD = $(THREAD_LIBS) $(DVDREAD_LIBS)
 
 pkgincludedir = $(includedir)/dvdnav
-- 
2.11.0

