From 1776d9054e428d1e6ab7aee79cb05474355fdbbf Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 4 Jan 2019 15:32:34 +0100
Subject: [PATCH 2/2] Disable split-stack for non-thread builds

---
 libgcc/config/t-stack | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libgcc/config/t-stack b/libgcc/config/t-stack
index cc0366b..f3f97e8 100644
--- a/libgcc/config/t-stack
+++ b/libgcc/config/t-stack
@@ -1,4 +1,6 @@
 # Makefile fragment to provide generic support for -fsplit-stack.
 # This should be used in config.host for any host which supports
 # -fsplit-stack.
+ifeq ($(enable_threads),yes)
 LIB2ADD_ST += $(srcdir)/generic-morestack.c $(srcdir)/generic-morestack-thread.c
+endif
-- 
2.7.4

