From aec9c53549c9a755890114ccaacf5f15ed978dd9 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 9 Jan 2019 14:56:22 +0100
Subject: [PATCH 3/3] Force SEH+SJLJ

---
 libgcc/config.host | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libgcc/config.host b/libgcc/config.host
index 11b4aca..c76712c 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -765,10 +765,6 @@ x86_64-*-mingw*)
 	# This has to match the logic for DWARF2_UNWIND_INFO in gcc/config/i386/cygming.h
 	if test x$ac_cv_sjlj_exceptions = xyes; then
 		tmake_eh_file="i386/t-sjlj-eh"
-	elif test "${host_address}" = 32; then
-	        # biarch -m32 with --disable-sjlj-exceptions
-	     	tmake_eh_file="i386/t-dw2-eh"
-		md_unwind_header=i386/w32-unwind.h
 	else
 		tmake_eh_file="i386/t-seh-eh"
 	fi
-- 
2.7.4

