From 5988e14251cba4cfefabb9bf333fd5b2539cbc5d Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Wed, 21 Feb 2018 15:02:58 +0100
Subject: [PATCH] define IID_Unknown

---
 libass/dwrite_c.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libass/dwrite_c.h b/libass/dwrite_c.h
index c902525..d88513a 100644
--- a/libass/dwrite_c.h
+++ b/libass/dwrite_c.h
@@ -13,6 +13,7 @@
 #define DWRITEAPI DECLSPEC_IMPORT
 
 #include <unknwn.h>
+DEFINE_GUID(IID_IUnknown, 0x00000000, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
 
 typedef struct IDWriteFactory IDWriteFactory;
 typedef struct IDWriteFont IDWriteFont;
-- 
2.14.3

