From 8d10989ca6c3121726fce7143d7efec3f5387896 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sun, 26 Feb 2012 02:41:19 +0100
Subject: Fix include dir in .pc files


diff --git a/amrnb/opencore-amrnb.pc.in b/amrnb/opencore-amrnb.pc.in
index b199fca..b838c0d 100644
--- a/amrnb/opencore-amrnb.pc.in
+++ b/amrnb/opencore-amrnb.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/opencore-amrnb
 
 Name: OpenCORE AMR-NB
 Description: Adaptive Multi-Rate speech codec library
diff --git a/amrwb/opencore-amrwb.pc.in b/amrwb/opencore-amrwb.pc.in
index b4eb993..8446310 100644
--- a/amrwb/opencore-amrwb.pc.in
+++ b/amrwb/opencore-amrwb.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/opencore-amrwb
 
 Name: OpenCORE AMR-WB
 Description: Adaptive Multi-Rate Wideband speech codec library
-- 
cgit v0.10.2


