From c22383a042c424c946d8dfb1843d6dd4135d784f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Fri, 9 Dec 2016 15:24:04 +0200
Subject: [PATCH 2/2] Build DLLs on Windows

---
 Makefile.am  | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e51764c..f92ae8b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -33,7 +33,7 @@ libdvdread_la_SOURCES = \
 
 libdvdread_la_LIBADD = $(CSS_LIBS)
 
-libdvdread_la_LDFLAGS = -version-info $(DVDREAD_LTVERSION) \
+libdvdread_la_LDFLAGS = -no-undefined -version-info $(DVDREAD_LTVERSION) \
 	-export-symbols-regex "(^dvd.*|^nav.*|^ifo.*|^DVD.*|^UDF.*)"
 
 pkgincludedir = $(includedir)/dvdread
diff --git a/configure.ac b/configure.ac
index 7b2dad3..ed91829 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,7 +9,7 @@ AM_MAINTAINER_MODE([enable])
 dnl Enable silent rules only when available (automake 1.11 or later).
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-LT_INIT
+LT_INIT([win32-dll])
 
 AC_CONFIG_HEADER(config.h)
 AC_CONFIG_MACRO_DIR([m4])
-- 
2.11.0

