diff -ur gtk-sharp-2.12.10.modif/glib/glue/list.c gtk-sharp-2.12.10/glib/glue/list.c
--- gtk-sharp-2.12.10.modif/glib/glue/list.c	2012-08-02 17:50:06.000000000 +0200
+++ gtk-sharp-2.12.10/glib/glue/list.c	2009-01-07 17:54:06.000000000 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
diff -ur gtk-sharp-2.12.10.modif/glib/glue/slist.c gtk-sharp-2.12.10/glib/glue/slist.c
--- gtk-sharp-2.12.10.modif/glib/glue/slist.c	2012-08-02 17:50:06.000000000 +0200
+++ gtk-sharp-2.12.10/glib/glue/slist.c	2009-01-07 17:54:06.000000000 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
diff -ur gtk-sharp-2.12.10.modif/glib/glue/thread.c gtk-sharp-2.12.10/glib/glue/thread.c
--- gtk-sharp-2.12.10.modif/glib/glue/thread.c	2012-08-02 17:50:06.000000000 +0200
+++ gtk-sharp-2.12.10/glib/glue/thread.c	2009-01-07 17:54:06.000000000 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gthread.h>
+#include <glib.h>
 
 gboolean glibsharp_g_thread_supported (void);
