From 04b4406054ca9ffb5d3edf9206e5cec1529450b9 Mon Sep 17 00:00:00 2001
From: Josep Torra <n770galaxy@gmail.com>
Date: Mon, 12 Mar 2012 11:00:42 +0100
Subject: [PATCH 06/16] Fix build in PPC.

---
 libvisual/lv_cpu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libvisual/lv_cpu.c b/libvisual/lv_cpu.c
index fe09752..b6349a0 100644
--- a/libvisual/lv_cpu.c
+++ b/libvisual/lv_cpu.c
@@ -144,7 +144,9 @@ static void sigill_handler (int sig)
 	__lv_powerpc_canjump = 0;
 	siglongjmp (__lv_powerpc_jmpbuf, 1);
 }
+#endif
 
+#if defined(VISUAL_ARCH_POWERPC)
 static void check_os_altivec_support( void )
 {
 #if defined(VISUAL_OS_DARWIN)
-- 
2.1.4

