From 567baa726cf34d99a27c24ec62c37b94475a195d Mon Sep 17 00:00:00 2001
From: "Andre Moreira Magalhaes (andrunko)" <andre.magalhaes@collabora.co.uk>
Date: Mon, 30 Apr 2012 22:43:01 +0100
Subject: [PATCH 07/16] Fix build on debian squeeze.

---
 libvisual/lv_cpu.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libvisual/lv_cpu.c b/libvisual/lv_cpu.c
index b6349a0..123824a 100644
--- a/libvisual/lv_cpu.c
+++ b/libvisual/lv_cpu.c
@@ -76,7 +76,7 @@ static int cpuid (unsigned int ax, unsigned int *p);
 
 /* The sigill handlers */
 #if defined(VISUAL_ARCH_X86) //x86 (linux katmai handler check thing)
-#if defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE) && defined(X86_FXSR_MAGIC)
+#if defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE)
 static void sigill_handler_sse( int signal, struct sigcontext sc )
 {
 	/* Both the "xorps %%xmm0,%%xmm0" and "divps %xmm0,%%xmm1"
@@ -109,7 +109,7 @@ static void sigfpe_handler_sse( int signal, struct sigcontext sc )
 	}
 }
 #endif
-#endif /* VISUAL_OS_LINUX && _POSIX_SOURCE && X86_FXSR_MAGIC */
+#endif /* VISUAL_OS_LINUX && _POSIX_SOURCE */
 
 #if defined(VISUAL_OS_WIN32)
 LONG CALLBACK win32_sig_handler_sse(EXCEPTION_POINTERS* ep)
@@ -149,6 +149,7 @@ static void sigill_handler (int sig)
 #if defined(VISUAL_ARCH_POWERPC)
 static void check_os_altivec_support( void )
 {
+  return;
 #if defined(VISUAL_OS_DARWIN)
 	int sels[2] = {CTL_HW, HW_VECTORUNIT};
 	int has_vu = 0;
-- 
2.1.4

