From 2c6cabc6012de430556d4d1f2be20b2ac064b9f4 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 3 Jan 2013 14:32:47 +0100
Subject: [PATCH 09/16] Fix build for Android X86

---
 libvisual/lv_cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libvisual/lv_cpu.c b/libvisual/lv_cpu.c
index 123824a..2951dc2 100644
--- a/libvisual/lv_cpu.c
+++ b/libvisual/lv_cpu.c
@@ -76,7 +76,7 @@ static int cpuid (unsigned int ax, unsigned int *p);
 
 /* The sigill handlers */
 #if defined(VISUAL_ARCH_X86) //x86 (linux katmai handler check thing)
-#if defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE)
+#if defined(VISUAL_OS_LINUX) && (defined(_POSIX_SOURCE) || defined(__BIONIC__))
 static void sigill_handler_sse( int signal, struct sigcontext sc )
 {
 	/* Both the "xorps %%xmm0,%%xmm0" and "divps %xmm0,%%xmm1"
-- 
2.1.4

