From 99d4031733eb30e614b4374b441102781315d065 Mon Sep 17 00:00:00 2001
From: Olivier Crete <olivier.crete@collabora.com>
Date: Fri, 27 Apr 2018 14:19:45 -0400
Subject: [PATCH 2/2] ios: Remove flags incompatible with -fembed-bitcode

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index c0a67be..315bebf 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -98,8 +98,8 @@ SHAREDLIBSUFFIXNOVER = dll
 SHAREDLIBSUFFIX = $(SHAREDLIBSUFFIXNOVER)
 else ifeq (darwin,$(findstring darwin,@host@))
 SHAREDLIB_DIR = $(libdir)
-SHAREDLIB_LDFLAGS = -dynamiclib -twolevel_namespace -undefined dynamic_lookup \
-        -fno-common -headerpad_max_install_names -install_name $(libdir)/$@
+SHAREDLIB_LDFLAGS = -dynamiclib -twolevel_namespace \
+        -fno-common -install_name $(libdir)/$@
 SHAREDLIBSUFFIXNOVER = dylib
 SHAREDLIBSUFFIX = $(SHAREDLIBVERSION).$(SHAREDLIBSUFFIXNOVER)
 endif
-- 
2.15.1 (Apple Git-101)

