From 8f309471cdcddc18f96ebdc97526150706ac119d Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@gnome.org>
Date: Wed, 25 Feb 2015 12:16:26 +0100
Subject: [PATCH] configure: Set gettext to version 0.19

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f5d3bc6..b11486a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -175,7 +175,7 @@ AC_SUBST(MAINTAINER_CFLAGS)
 
 dnl = I18N =================================================================
 
-AM_GNU_GETTEXT_VERSION([0.18])
+AM_GNU_GETTEXT_VERSION([0.19])
 AM_GNU_GETTEXT([external])
 
 GETTEXT_PACKAGE="cluttergtk-1.0"
-- 
2.1.0

