From c2208acb3393eda4b5dc266328474bd918f2039d Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 8 Mar 2012 17:56:16 +0100
Subject: [PATCH 2/3] Fix windows cross-compilation

---
 configure.ac    | 37 ++++++++++++++++++++++++++++++++++++-
 src/Makefile.am |  9 +++++++--
 2 files changed, 43 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 02158de..71e9d6a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,8 +53,43 @@ PKG_CHECK_MODULES(XPYB, xpyb >= xpyb_required_version,[have_xpyb=yes],[have_xpyb
 AS_IF([test "x$have_xpyb" = "xyes"],
       [AC_DEFINE([HAVE_XPYB], [1], [Define to 1 if xpyb is available])])
 
+AC_MSG_CHECKING([for some Win32 platform])
+case "$host" in
+  *-*-mingw*|*-*-cygwin*)
+    platform_win32=yes
+    ;;
+  *)
+    platform_win32=no
+    ;;
+esac
+AC_MSG_RESULT([$platform_win32])
+
+AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
 # Checks for header files ---
-AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
+if test "x$cross_compiling" = "xyes" -a "x$platform_win32" = "xyes" ; then
+  # skip python cheks if cross-compiling to windows
+  # $PYTHON_INCLUDES and PYTHON_LIBS should be set properly
+  save_CPPFLAGS="$CPPFLAGS"
+  CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+  /* end confdefs.h.  */
+  #include <Python.h>
+_ACEOF
+  if ac_fn_c_try_cpp "$LINENO"; then :
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
+      $as_echo "found" >&6; }
+
+  else
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: not found" >&5
+      $as_echo "not found" >&6; }
+    as_fn_error $? "could not find Python headers" "$LINENO" 5
+   fi
+   rm -f conftest.err conftest.i conftest.$ac_ext
+   CPPFLAGS="$save_CPPFLAGS"
+   AC_SUBST(PYTHON_LIBS)
+else
+  AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
+fi
 
 # checks for types and compilers ---
 AC_C_CONST
diff --git a/src/Makefile.am b/src/Makefile.am
index 64e7b29..87b7634 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,6 +1,11 @@
 EXTRA_DIST = \
   wscript
 
+ldflags = -module -avoid-version -export-symbols-regex init_cairo
+if PLATFORM_WIN32
+ldflags += -no-undefined
+endif
+
 pycairoexecdir = $(pyexecdir)/cairo
 
 pkgincludedir = $(includedir)/pycairo
@@ -9,8 +14,8 @@ pkginclude_HEADERS = pycairo.h
 pycairoexec_PYTHON = __init__.py
 
 pycairoexec_LTLIBRARIES = _cairo.la
-_cairo_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_cairo
-_cairo_la_LIBADD = $(CAIRO_LIBS)
+_cairo_la_LDFLAGS = $(ldflags)
+_cairo_la_LIBADD = $(PYTHON_LIBS) $(CAIRO_LIBS)
 _cairo_la_CPPFLAGS = $(PYTHON_INCLUDES) $(CAIRO_CFLAGS)
 _cairo_la_SOURCES = \
   cairomodule.c \
-- 
2.1.4

