From 9d4196a8868abd07dfb1ce9a3fb1dda23872061e Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Thu, 29 Jan 2015 15:37:02 +0100
Subject: [PATCH] libffi: Don't be smart about toolexeclibdir

---
 configure.ac | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/configure.ac b/configure.ac
index a7bf5ee..fe6f905 100644
--- a/configure.ac
+++ b/configure.ac
@@ -590,26 +590,7 @@ AC_ARG_ENABLE(purify-safety,
     AC_DEFINE(USING_PURIFY, 1, [Define this if you are using Purify and want to suppress spurious messages.])
   fi)
 
-# These variables are only ever used when we cross-build to X86_WIN32.
-# And we only support this with GCC, so...
-if test "x$GCC" = "xyes"; then
-  if test -n "$with_cross_host" &&
-     test x"$with_cross_host" != x"no"; then
-    toolexecdir="${exec_prefix}"/'$(target_alias)'
-    toolexeclibdir="${toolexecdir}"/lib
-  else
-    toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
-    toolexeclibdir="${libdir}"
-  fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
-  AC_SUBST(toolexecdir)
-else
-  toolexeclibdir="${libdir}"
-fi
+toolexeclibdir="${libdir}"
 AC_SUBST(toolexeclibdir)
 
 AC_CONFIG_COMMANDS(include, [test -d include || mkdir include])
-- 
2.1.0

