From 9c12e94c2602c9d2698c0d6104ba7d5cd75dfa10 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 28 Oct 2013 17:45:44 +0100
Subject: [PATCH 1/2] arm: fix alignment

---
 src/arm/sysv.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/arm/sysv.S b/src/arm/sysv.S
index fb38cd6..95b8b27 100644
--- a/src/arm/sysv.S
+++ b/src/arm/sysv.S
@@ -113,7 +113,7 @@
 #if defined(__thumb__) && !defined(__THUMB_INTERWORK__)
 .macro	ARM_FUNC_START name
 	.text
-	.align 0
+	.align 2
 	.thumb
 	.thumb_func
 #ifdef __APPLE__
@@ -136,7 +136,7 @@ _L__\name:
 #else
 .macro	ARM_FUNC_START name
 	.text
-	.align 0
+	.align 2
 	.arm
 #ifdef __APPLE__
 	ENTRY($0)
-- 
1.8.1.2

