--- libffi-3.0.13/src/arm/sysv.S.old	2013-09-25 16:27:59.000000000 +0200
+++ libffi-3.0.13/src/arm/sysv.S	2013-09-25 16:28:05.000000000 +0200
@@ -113,7 +113,7 @@
 #if defined(__thumb__) && !defined(__THUMB_INTERWORK__)
 .macro	ARM_FUNC_START name
 	.text
-	.align 0
+	.align 2
 	.thumb
 	.thumb_func
 #ifdef __APPLE__
@@ -136,7 +136,7 @@
 #else
 .macro	ARM_FUNC_START name
 	.text
-	.align 0
+	.align 2
 	.arm
 #ifdef __APPLE__
 	ENTRY($0)

