From d4518b3f184d538c03c3c15d3b64b5a41ebfa529 Mon Sep 17 00:00:00 2001
From: Thomas Coldrick <othko97@gmail.com>
Date: Thu, 29 Aug 2019 15:09:50 +0100
Subject: [PATCH] meson.build: Bump soname version to 5

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 8813deb8a..72a36a4cc 100644
--- a/meson.build
+++ b/meson.build
@@ -1,10 +1,10 @@
 project('openh264', ['c', 'cpp'],
-  version : '1.8.0',
+  version : '2.0.0',
   meson_version : '>= 0.43',
   default_options : [ 'warning_level=1',
                       'buildtype=debugoptimized' ])
 
-major_version = '4'
+major_version = '5'
 
 cpp = meson.get_compiler('cpp')
 
