From 2fa9af08f511f2ff8e12e894d08bd492f027ac13 Mon Sep 17 00:00:00 2001
From: Olivier Crete <olivier.crete@collabora.com>
Date: Fri, 27 Apr 2018 13:56:15 -0400
Subject: [PATCH 1/2] openh264: Build for 8.0 -fembed-bitcode

---
 build/platform-ios.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/platform-ios.mk b/build/platform-ios.mk
index 4f219e3..37d43a6 100644
--- a/build/platform-ios.mk
+++ b/build/platform-ios.mk
@@ -7,7 +7,7 @@ SDKTYPE = iPhoneSimulator
 else
 SDKTYPE = iPhoneOS
 endif
-SDK_MIN = 5.1
+SDK_MIN = 8.0
 
 SDKROOT = $(shell xcrun --sdk $(shell echo $(SDKTYPE) | tr A-Z a-z) --show-sdk-path)
 CFLAGS += -arch $(ARCH) -isysroot $(SDKROOT) -miphoneos-version-min=$(SDK_MIN) -DAPPLE_IOS -fembed-bitcode
-- 
2.15.1 (Apple Git-101)

