From 155a641c3fc4787d381ba15692f99d61e3bd9b53 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Thu, 26 Mar 2020 11:00:14 +1100
Subject: [PATCH] meson: disable tests automatically it gtest is not found

---
 test/meson.build | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/test/meson.build b/test/meson.build
index 4cda986..5be8eda 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -1,17 +1,19 @@
-if not get_option('tests').disabled()
-  gtest_main_dep = dependency('gtest', main : true, fallback: ['gtest', 'gtest_main_dep'], required: true)
-  gtest_dep = dependency('gtest', fallback: ['gtest', 'gtest_dep'], required: true)
+build_tests = get_option('tests')
+
+if not build_tests.disabled()
+  gtest_main_dep = dependency('gtest', main : true, fallback: ['gtest', 'gtest_main_dep'], required: build_tests)
+  gtest_dep = dependency('gtest', fallback: ['gtest', 'gtest_dep'], required: build_tests)
 
   test_inc = include_directories('.')
 
   test_data_generator_sources = files('api/DataGenerator.cpp')
   test_base_encoder_sources = files('api/BaseEncoderTest.cpp')
 
-  if gtest_dep.found()
+  if gtest_dep.found() and gtest_main_dep.found()
     subdir('api')
     subdir('common')
     subdir('decoder')
     subdir('encoder')
     subdir('processing')
   endif
-endif
\ No newline at end of file
+endif
-- 
2.25.2

