From 8aacdde0c49d70b0f5d80a618e1ee5b4683174a1 Mon Sep 17 00:00:00 2001
From: Olivier Crete <olivier.crete@collabora.com>
Date: Fri, 27 Apr 2018 14:05:16 -0400
Subject: [PATCH] ios: Remove flags incompatible with -fembed-bitcode

---
 build/platform-darwin.mk | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/build/platform-darwin.mk b/build/platform-darwin.mk
index 6c7b244..9629471 100644
--- a/build/platform-darwin.mk
+++ b/build/platform-darwin.mk
@@ -5,8 +5,7 @@ SHAREDLIBSUFFIXFULLVER=$(FULL_VERSION).$(SHAREDLIBSUFFIX)
 SHAREDLIBSUFFIXMAJORVER=$(SHAREDLIB_MAJORVERSION).$(SHAREDLIBSUFFIX)
 CURRENT_VERSION := 2.0.0
 COMPATIBILITY_VERSION := 2.0.0
-SHLDFLAGS = -dynamiclib -twolevel_namespace -undefined dynamic_lookup \
-	-fno-common -headerpad_max_install_names -install_name \
+SHLDFLAGS = -dynamiclib -twolevel_namespace -fno-common -install_name \
 	$(SHAREDLIB_DIR)/$(LIBPREFIX)$(PROJECT_NAME).$(SHAREDLIBSUFFIXMAJORVER)
 SHARED = -dynamiclib
 SHARED += -current_version $(CURRENT_VERSION) -compatibility_version $(COMPATIBILITY_VERSION)
-- 
2.25.1

