From 36d2dcadb1884938f23aa349a056f92f13d09961 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 30 Aug 2012 21:14:07 +0200
Subject: [PATCH 2/2] fix autoreconf

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index 451921d..17f8c7b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -30,6 +30,7 @@
 #
 
 ACLOCAL_AMFLAGS = -I m4
+AUTOMAKE_OPTIONS = foreign 1.7
 
 SUBDIRS = doc include m4 man src examples test build objs
 
-- 
1.9.0

