From 8f5d2c0283c4cbc430ac30a6efef999c33bdb2c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Cerveau?= <scerveau@igalia.com>
Date: Fri, 4 Nov 2022 12:20:25 +0100
Subject: [PATCH] pkgconfig: fix generation to use prefix

---
 ffbuild/pkgconfig_generate.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ffbuild/pkgconfig_generate.sh b/ffbuild/pkgconfig_generate.sh
index e5de671..679413e 100755
--- a/ffbuild/pkgconfig_generate.sh
+++ b/ffbuild/pkgconfig_generate.sh
@@ -29,8 +29,8 @@ version=$(grep ${name}_VERSION= $name/${name}.version | cut -d= -f2)
 cat <<EOF > $name/$fullname.pc
 prefix=$prefix
 exec_prefix=\${prefix}
-libdir=$libdir
-includedir=$incdir
+libdir=\${prefix}/lib
+includedir=\${prefix}/include
 
 Name: $fullname
 Description: $comment
-- 
2.34.1

