From 7da1af870dad6a6f2f781f8cddbc061755492b06 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Fri, 26 Oct 2012 03:01:13 +0200
Subject: [PATCH] be more permissive with automake errors now that we use
 automake 1.12

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 58ff478..2bbf6dd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,7 +7,7 @@ AC_PREREQ(2.53)
 
 AC_CANONICAL_TARGET
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
+AM_INIT_AUTOMAKE([-Wall foreign subdir-objects])
 AC_CONFIG_HEADERS([include/config.h])
 
 ifdef([AM_SILENT_RULES],[AM_SILENT_RULES])
diff --git a/Makefile.am b/Makefile.am
index 12a85d3..0123e0d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS=. tools tests doc
+SUBDIRS=. # tools tests doc
 
 OGGDIR=built-streams
 kateincdir=$(includedir)/kate
-- 
2.1.4

