From db06b9322de28af1c868a9d4dab2728f6515b996 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Wed, 24 Feb 2016 12:42:26 +0200
Subject: [PATCH 2/4] configure: Android is ELF too

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 22b2d98..5185daa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -96,6 +96,7 @@ use_accel=yes
 case "$host" in
   *android*)
     have_android=yes
+    have_elf=yes
   ;;
   *mingw32* | *mingw64*)
     have_win=yes
-- 
2.7.0

