From b450f470d17279c8ad0cf7c4df1f6fdbc911137a Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 31 Mar 2015 18:23:56 +0530
Subject: [PATCH 2/2] configure: Update gettext version to 0.19

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 40a01d7..87c3337 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,7 +150,7 @@ AM_CONDITIONAL(ENABLE_TESTS, test "$enable_tests" != "no")
 
 GTK_DOC_CHECK(1.1)
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.18])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 AC_C_BIGENDIAN
 
-- 
2.1.0

