From 0a61174452da1914a35f00513b5b7256208bca4e Mon Sep 17 00:00:00 2001
From: Olivier Crete <olivier.crete@collabora.com>
Date: Fri, 27 Apr 2018 14:12:57 -0400
Subject: [PATCH 2/6] ios: Remove flags incompatible with -fembed-bitcode

---
 librtmp/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/librtmp/Makefile b/librtmp/Makefile
index f4e2a0b..159f0e3 100644
--- a/librtmp/Makefile
+++ b/librtmp/Makefile
@@ -55,8 +55,8 @@ SODIR_mingw=$(BINDIR)
 SODIR=$(SODIR_$(SYS))
 
 SO_LDFLAGS_posix=-shared -Wl,-soname,$@
-SO_LDFLAGS_darwin=-dynamiclib -twolevel_namespace -undefined dynamic_lookup \
-	-fno-common -headerpad_max_install_names -install_name $(libdir)/$@
+SO_LDFLAGS_darwin=-dynamiclib -twolevel_namespace \
+	-fno-common -install_name $(libdir)/$@
 SO_LDFLAGS_mingw=-shared -Wl,--out-implib,librtmp.dll.a
 SO_LDFLAGS=$(SO_LDFLAGS_$(SYS))
 
-- 
2.24.1

