From 748e7f74c02b0cb52c708ff231704fdabb40baee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Wed, 14 Dec 2016 12:22:42 +0200
Subject: [PATCH] Include limits.h for SIZE_MAX

Otherwise compilation fails on Android.
---
 jmemmgr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/jmemmgr.c b/jmemmgr.c
index adccbdf..b17c5c5 100644
--- a/jmemmgr.c
+++ b/jmemmgr.c
@@ -33,6 +33,7 @@
 #include "jpeglib.h"
 #include "jmemsys.h"            /* import the system-dependent declarations */
 #include <stdint.h>
+#include <limits.h>
 
 #ifndef NO_GETENV
 #ifndef HAVE_STDLIB_H           /* <stdlib.h> should declare getenv() */
-- 
2.11.0

