From e6164509e6facfce5d7fad37d9454d1d0be6b127 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Tue, 22 Oct 2024 22:02:14 +0000
Subject: [PATCH 3/3] util: Disable quicserver with no-tests

---
 util/build.info | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/build.info b/util/build.info
index e4aab44..22d41bf 100644
--- a/util/build.info
+++ b/util/build.info
@@ -6,7 +6,7 @@ SCRIPTS{noinst}=wrap.pl
 SOURCE[wrap.pl]=wrap.pl.in
 DEPEND[wrap.pl]=../configdata.pm
 
-IF[{- !$disabled{quic} && !$disabled{stdio} && !$disabled{apps} -}]
+IF[{- !$disabled{quic} && !$disabled{stdio} && !$disabled{apps} && !$disabled{tests} -}]
   PROGRAMS{noinst}=quicserver
   SOURCE[quicserver]=quicserver.c
   INCLUDE[quicserver]=../include ../apps/include
-- 
2.46.0

