From f220eeb873699fe8492824b26d3f4ac1e4779df2 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 5 Apr 2012 03:13:02 +0200
Subject: [PATCH 1/5] Use native iconv in windows

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 3d0aba4..2d456a9 100644
--- a/configure.in
+++ b/configure.in
@@ -20,7 +20,7 @@ case $host in
         AC_DEFINE([BEOS],1,[Define if compiling on BeOS system.])
         ;;
     *mingw*)
-        MMS_LIBS="$MMS_LIBS -lws2_32"
+        MMS_LIBS="$MMS_LIBS -lws2_32 -liconv"
 dnl Needed for getaddrinfo under mingw
         MMS_CFLAGS="-DWINVER=0x0600 -D_WIN32_WINNT=0x0600"
         ;;
-- 
2.1.4

