From a5857c5a7e5f24b520d2aa98a9deb770835e1f98 Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward@centricular.com>
Date: Mon, 18 May 2015 11:49:09 +0200
Subject: [PATCH] uri: Add implementation of strndup for platforms that do not
 have it

Like mingw
---
 src/uri.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/uri.c b/src/uri.c
index a96b900..ff823eb 100644
--- a/src/uri.c
+++ b/src/uri.c
@@ -26,6 +26,25 @@
 #define TRUE 1
 #endif
 
+#ifndef strdup
+/**
+ * return a malloc'd copy of at most the specified
+ * number of bytes of a string
+ * @author glibc-Team
+ */
+char *strndup (const char *s, size_t n)
+{
+  size_t len = strnlen (s, n);
+  char *new = (char *) malloc (len + 1);
+
+  if (new == NULL)
+    return NULL;
+
+  new[len] = '\0';
+  return (char *) memcpy (new, s, len);
+}
+#endif
+
 #define g_strdup strdup
 #define g_strndup strndup
 #define g_malloc malloc
-- 
2.1.0

