From 7c9c7fdee52e8632b1866ffc52f711b20f335831 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Tue, 29 May 2012 13:07:02 +0200
Subject: [PATCH 5/5] Fix build on OSX leopard

---
 src/mms.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/mms.h b/src/mms.h
index 7e9a9c8..ea15fb9 100644
--- a/src/mms.h
+++ b/src/mms.h
@@ -22,6 +22,11 @@
  * libmms public header
  */
 
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
+
 #ifndef HAVE_MMS_H
 #define HAVE_MMS_H
 
-- 
2.1.4

