From 935d1879779b745750d8845c03d6bbebe2950eca Mon Sep 17 00:00:00 2001
From: Justin Kim <justin.kim@collabora.com>
Date: Fri, 23 Feb 2018 23:00:11 +0900
Subject: [PATCH] configure.ac: Don't check intltool version

Signed-off-by: Justin Kim <justin.kim@collabora.com>
---
 Makefile.am  | 2 +-
 configure.ac | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 38d2872..2363d4c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = libsoup po tests examples docs win32
+SUBDIRS = libsoup tests examples docs win32
 
 EXTRA_DIST =				\
 	data/effective_tld_names.dat	\
diff --git a/configure.ac b/configure.ac
index ace4add..20518e4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,7 +121,6 @@ AM_CONDITIONAL(OS_LINUX, [test $os_linux = yes])
 dnl ************************
 dnl *** gettext/intltool ***
 dnl ************************
-IT_PROG_INTLTOOL([0.35.0])
 GETTEXT_PACKAGE=libsoup
 AC_SUBST([GETTEXT_PACKAGE])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[The gettext domain name])
-- 
2.16.2

