From bb3cfc0e16e592e93bcda45ecab61ce5d9bc3009 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 1 Mar 2012 03:39:51 +0100
Subject: [PATCH 2/2] Don't build tests

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index a99287c..a7ddcf2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = libsoup po tests examples docs build/win32
+SUBDIRS = libsoup po examples docs build/win32
 
 EXTRA_DIST =				\
 	data/effective_tld_names.dat	\
-- 
2.11.0

