From 805e84b2400d0ec3b1fbfc12db5e3b7a9937dc45 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 6 Sep 2012 18:06:51 +0200
Subject: [PATCH 3/4] fix link using --with-thread=gthread

---
 schroedinger/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/schroedinger/Makefile.am b/schroedinger/Makefile.am
index 8987073..8e31457 100644
--- a/schroedinger/Makefile.am
+++ b/schroedinger/Makefile.am
@@ -154,6 +154,7 @@ libschroedinger_@SCHRO_MAJORMINOR@_la_LIBADD = \
 	$(OPENGL_LIBS) \
 	$(ORC_LIBS) \
 	$(GCOV_LIBS) \
+	$(GTHREAD_LIBS) \
 	$(PTHREAD_LIBS)
 libschroedinger_@SCHRO_MAJORMINOR@_la_LDFLAGS = \
 	-no-undefined \
-- 
2.1.4

