From 3549c1131ba5fdceaed575c923d145e411281e81 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 12 Sep 2012 17:56:23 +0200
Subject: [PATCH 2/7] fix android build with stlport

---
 CMakeLists.txt        | 17 +++++++++++++++--
 taglib/CMakeLists.txt |  4 ++++
 2 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5194af..82b3539 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,8 +24,13 @@ set(INCLUDE_INSTALL_DIR      "${CMAKE_INSTALL_PREFIX}/include"             CACHE
 
 if (CMAKE_COMPILER_IS_GNUCXX)
    if (CMAKE_SYSTEM_NAME MATCHES Linux)
-     set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -ansi -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
-     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -fno-check-new -fno-common")
+     set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
+     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long  -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -fno-check-new -fno-common")
+     if (ANDROID_NDK)
+     else (ANDROID_NDK)
+       set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS}  ")
+       set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  ")
+     endif (ANDROID_NDK)
    endif (CMAKE_SYSTEM_NAME MATCHES Linux)
 endif (CMAKE_COMPILER_IS_GNUCXX)
 if(MSVC)
@@ -38,6 +43,14 @@ if (WIN32)
   set(CMAKE_DEBUG_POSTFIX "d")
 endif (WIN32)
 
+if (ANDROID_NDK)
+    include(FindPkgConfig)
+    pkg_check_modules (STLPORT stlport)
+    include_directories(${STLPORT_INCLUDE_DIRS}  )
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${STLPORT_CFLAGS_OTHER}  ")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${STLPORT_CFLAGS_OTHER} ")
+endif (ANDROID_NDK)
+
 SET(TAGLIB_LIB_MAJOR_VERSION "1")
 SET(TAGLIB_LIB_MINOR_VERSION "7")
 SET(TAGLIB_LIB_PATCH_VERSION "2")
diff --git a/taglib/CMakeLists.txt b/taglib/CMakeLists.txt
index f050a5c..9c4b7f1 100644
--- a/taglib/CMakeLists.txt
+++ b/taglib/CMakeLists.txt
@@ -203,6 +203,10 @@ if(ZLIB_FOUND)
 	TARGET_LINK_LIBRARIES(tag ${ZLIB_LIBRARIES})
 endif(ZLIB_FOUND)
 
+if(ANDROID_NDK)
+	TARGET_LINK_LIBRARIES(tag  ${STLPORT_STATIC_LDFLAGS_OTHER})
+endif(ANDROID_NDK)
+
 SET_TARGET_PROPERTIES(tag PROPERTIES
         VERSION ${TAGLIB_LIB_MAJOR_VERSION}.${TAGLIB_LIB_MINOR_VERSION}.${TAGLIB_LIB_PATCH_VERSION}
         SOVERSION ${TAGLIB_LIB_MAJOR_VERSION}
-- 
2.1.4

