From dcf4e27fe827b9bf21a1e896378597391ee5f251 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 12 Sep 2012 18:14:11 +0200
Subject: [PATCH 3/7] taglib.pc: require stlport for android

---
 CMakeLists.txt  | 1 +
 taglib.pc.cmake | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82b3539..692a1e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,6 +46,7 @@ endif (WIN32)
 if (ANDROID_NDK)
     include(FindPkgConfig)
     pkg_check_modules (STLPORT stlport)
+    set (STLPORT_PC "stlport")
     include_directories(${STLPORT_INCLUDE_DIRS}  )
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${STLPORT_CFLAGS_OTHER}  ")
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${STLPORT_CFLAGS_OTHER} ")
diff --git a/taglib.pc.cmake b/taglib.pc.cmake
index 909b8fc..4665641 100644
--- a/taglib.pc.cmake
+++ b/taglib.pc.cmake
@@ -5,7 +5,7 @@ includedir=${INCLUDE_INSTALL_DIR}
 
 Name: TagLib
 Description: Audio meta-data library
-Requires: 
+Requires:  ${STLPORT_PC}
 Version: ${TAGLIB_LIB_MAJOR_VERSION}.${TAGLIB_LIB_MINOR_VERSION}.${TAGLIB_LIB_PATCH_VERSION}
 Libs: -L${LIB_INSTALL_DIR} -ltag
 Cflags: -I${INCLUDE_INSTALL_DIR}/taglib 
-- 
2.1.4

