From 0aa3e6d2e00d2fa540664b1473d8ae18b3ee3408 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Wed, 3 Oct 2012 17:25:26 +0200
Subject: [PATCH 4/7] build shared and static libraries

---
 taglib/CMakeLists.txt | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/taglib/CMakeLists.txt b/taglib/CMakeLists.txt
index 9c4b7f1..d682667 100644
--- a/taglib/CMakeLists.txt
+++ b/taglib/CMakeLists.txt
@@ -191,12 +191,9 @@ SET(tag_LIB_SRCS ${mpeg_SRCS} ${id3v1_SRCS} ${id3v2_SRCS} ${frames_SRCS} ${ogg_S
 )
 
 
-if(ENABLE_STATIC)
-    add_library(tag STATIC ${tag_LIB_SRCS})
-    set_target_properties(tag PROPERTIES COMPILE_DEFINITIONS TAGLIB_STATIC)
-else(ENABLE_STATIC)
-    add_library(tag SHARED ${tag_LIB_SRCS})
-endif(ENABLE_STATIC)
+add_library(tag_static STATIC ${tag_LIB_SRCS})
+set_target_properties(tag_static PROPERTIES COMPILE_DEFINITIONS TAGLIB_STATIC)
+add_library(tag SHARED ${tag_LIB_SRCS})
 
 TARGET_LINK_LIBRARIES(tag )
 if(ZLIB_FOUND)
@@ -214,7 +211,7 @@ SET_TARGET_PROPERTIES(tag PROPERTIES
         DEFINE_SYMBOL MAKE_TAGLIB_LIB
         LINK_INTERFACE_LIBRARIES ""
 )
-INSTALL(TARGETS tag
+INSTALL(TARGETS tag tag_static
 	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
 	RUNTIME DESTINATION ${BIN_INSTALL_DIR}
 	ARCHIVE DESTINATION  ${LIB_INSTALL_DIR}
-- 
2.1.4

