From c3129953f17cad928e74582fa697ba3342d465aa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Thu, 25 Oct 2012 10:52:57 +0200
Subject: [PATCH 5/7] Use gnustl instead of stlport

---
 CMakeLists.txt        | 10 +++++-----
 taglib/CMakeLists.txt |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 692a1e4..8142895 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,11 +45,11 @@ endif (WIN32)
 
 if (ANDROID_NDK)
     include(FindPkgConfig)
-    pkg_check_modules (STLPORT stlport)
-    set (STLPORT_PC "stlport")
-    include_directories(${STLPORT_INCLUDE_DIRS}  )
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${STLPORT_CFLAGS_OTHER}  ")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${STLPORT_CFLAGS_OTHER} ")
+    pkg_check_modules (GNUSTL gnustl)
+    set (GNUSTL_PC "gnustl")
+    include_directories(${GNUSTL_INCLUDE_DIRS}  )
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GNUSTL_CFLAGS_OTHER}  ")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${GNUSTL_CFLAGS_OTHER} ")
 endif (ANDROID_NDK)
 
 SET(TAGLIB_LIB_MAJOR_VERSION "1")
diff --git a/taglib/CMakeLists.txt b/taglib/CMakeLists.txt
index d682667..6358657 100644
--- a/taglib/CMakeLists.txt
+++ b/taglib/CMakeLists.txt
@@ -201,7 +201,7 @@ if(ZLIB_FOUND)
 endif(ZLIB_FOUND)
 
 if(ANDROID_NDK)
-	TARGET_LINK_LIBRARIES(tag  ${STLPORT_STATIC_LDFLAGS_OTHER})
+	TARGET_LINK_LIBRARIES(tag  ${GNUSTL_STATIC_LDFLAGS_OTHER})
 endif(ANDROID_NDK)
 
 SET_TARGET_PROPERTIES(tag PROPERTIES
-- 
2.1.4

