From a092425ed89444410bd61782f536d6fc584dc010 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Thu, 25 Oct 2012 11:44:26 +0200
Subject: [PATCH 7/7] Fix pkg-config file

---
 taglib.pc.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/taglib.pc.cmake b/taglib.pc.cmake
index 4665641..ebd5b15 100644
--- a/taglib.pc.cmake
+++ b/taglib.pc.cmake
@@ -5,7 +5,7 @@ includedir=${INCLUDE_INSTALL_DIR}
 
 Name: TagLib
 Description: Audio meta-data library
-Requires:  ${STLPORT_PC}
+Requires:  ${GNUSTL_PC}
 Version: ${TAGLIB_LIB_MAJOR_VERSION}.${TAGLIB_LIB_MINOR_VERSION}.${TAGLIB_LIB_PATCH_VERSION}
 Libs: -L${LIB_INSTALL_DIR} -ltag
 Cflags: -I${INCLUDE_INSTALL_DIR}/taglib 
-- 
2.1.4

