From a1ed37d405289702eecabeadf30ae0ad8f2b0731 Mon Sep 17 00:00:00 2001
From: Fernando Jimenez <fjimenez@igalia.com>
Date: Tue, 16 Feb 2021 20:56:10 +0100
Subject: [PATCH] Android logging macro

---
 Source/WebKit/config.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Source/WebKit/config.h b/Source/WebKit/config.h
index 8748a4e1..0dab2799 100644
--- a/Source/WebKit/config.h
+++ b/Source/WebKit/config.h
@@ -64,3 +64,6 @@
 #if ENABLE(WEBDRIVER_MOUSE_INTERACTIONS) || ENABLE(WEBDRIVER_KEYBOARD_INTERACTIONS) || ENABLE(WEBDRIVER_TOUCH_INTERACTIONS)
 #define ENABLE_WEBDRIVER_ACTIONS_API 1
 #endif
+
+#include <android/log.h>
+#define ALOGV(...) __android_log_print(ANDROID_LOG_VERBOSE, "WPEDebug", __VA_ARGS__)
-- 
2.25.1

