From 4d9121d87aab4d0866f4c570e26ed5a011b32d09 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fernando=20Jim=C3=A9nez=20Moreno?= <ferjmoreno@gmail.com>
Date: Tue, 25 May 2021 15:32:57 +0200
Subject: [PATCH 01/12] Android logging macro

---
 Source/WebKit/config.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Source/WebKit/config.h b/Source/WebKit/config.h
index 03f33943875e..cf0a70bcce8a 100644
--- a/Source/WebKit/config.h
+++ b/Source/WebKit/config.h
@@ -47,3 +47,6 @@
 #if ENABLE(WEBDRIVER_MOUSE_INTERACTIONS) || ENABLE(WEBDRIVER_KEYBOARD_INTERACTIONS) || ENABLE(WEBDRIVER_TOUCH_INTERACTIONS) || ENABLE(WEBDRIVER_WHEEL_INTERACTIONS)
 #define ENABLE_WEBDRIVER_ACTIONS_API 1
 #endif
+
+#include <android/log.h>
+#define ALOGV(...) __android_log_print(ANDROID_LOG_VERBOSE, "WPEDebug", __VA_ARGS__)
-- 
2.43.0

