From 5ebe77ac071f728ba9fde362620dcb200f8fe6ed Mon Sep 17 00:00:00 2001
From: Cerbero Build System <cerbero@gstreamer.freedesktop.org>
Date: Wed, 31 Mar 2021 15:41:38 +0200
Subject: [PATCH xserver 8/9] Try to find libz with pkg-config

---
 Source/cmake/WebKitFindPackage.cmake | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Source/cmake/WebKitFindPackage.cmake b/Source/cmake/WebKitFindPackage.cmake
index 8a0a1868c496..39b3b35e2481 100644
--- a/Source/cmake/WebKitFindPackage.cmake
+++ b/Source/cmake/WebKitFindPackage.cmake
@@ -83,6 +83,17 @@ macro(find_package package)
         else ()
            set(_found_package OFF)
         endif ()
+    else ()
+        if ("${package}" STREQUAL "ZLIB")
+          find_package(PkgConfig)
+          pkg_check_modules(PC_ZLIB zlib)
+          set(ZLIB_INCLUDE_DIRS ${PC_ZLIB_INCLUDE_DIRS})
+          set(ZLIB_LIBRARY ${PC_ZLIB_LINK_LIBRARIES})
+          if (ZLIB_INCLUDE_DIRS AND ZLIB_LIBRARY)
+            set(ZLIB_FOUND ON)
+            set(_found_package ON)
+          endif ()
+        endif ()
     endif ()
 
     # Apple builds have a unique location for ICU
-- 
2.30.2

