From afd19dbe8ec91ac4f7a4d0825137981d21ff8cae Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 15 Jan 2013 12:19:49 +0100
Subject: [PATCH 2/3] Fix includes for Bionic X86

---
 lib/libtasn1.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libtasn1.h b/lib/libtasn1.h
index 284a8e0..0f21db0 100644
--- a/lib/libtasn1.h
+++ b/lib/libtasn1.h
@@ -35,9 +35,9 @@
 #  endif
 # endif
 
-#include <stdio.h>		/* for FILE* */
 #include <sys/types.h>
 #include <time.h>
+#include <stdio.h>		/* for FILE* */
 
 #ifdef __cplusplus
 extern "C"
-- 
1.8.3.1

