From bfa09fd02e06f0fce4b9393d791c29b71e55dec2 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 15 Jan 2013 13:00:45 +0100
Subject: [PATCH 3/3] Fix check for Android x86

---
 gl/stdint.in.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gl/stdint.in.h b/gl/stdint.in.h
index e2a0eb1..2f80268 100644
--- a/gl/stdint.in.h
+++ b/gl/stdint.in.h
@@ -39,7 +39,7 @@
    Ideally we should test __BIONIC__ here, but it is only defined after
    <sys/cdefs.h> has been included; hence test __ANDROID__ instead.  */
 #if defined __ANDROID__ \
-    && defined _SYS_TYPES_H_ && !defined _SSIZE_T_DEFINED_
+    && defined _SYS_TYPES_H_ && !defined __need_size_t
 # @INCLUDE_NEXT@ @NEXT_STDINT_H@
 #else
 
-- 
1.8.3.1

