From 14d856db55054662db9f8b086cf22b19b094fbd5 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Mon, 28 Oct 2013 13:26:35 +0100
Subject: [PATCH 1/2] neon: fix function export names for iOS armv7

---
 arm/filter_neon.S | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/arm/filter_neon.S b/arm/filter_neon.S
index 2308aad..9e957e4 100644
--- a/arm/filter_neon.S
+++ b/arm/filter_neon.S
@@ -22,6 +22,12 @@
 
 #ifdef PNG_READ_SUPPORTED
 
+#if defined(__APPLE__)
+#define FUNC_PREFIX _
+#else
+#define FUNC_PREFIX
+#endif
+
 /* Assembler NEON support - only works for 32-bit ARM (i.e. it does not work for
  * ARM64).  The code in arm/filter_neon_intrinsics.c supports ARM64, however it
  * only works if -mfpu=neon is specified on the GCC command line.  See pngpriv.h
@@ -42,7 +48,7 @@
 
 .macro  func    name, export=0
     .macro endfunc
-ELF     .size   \name, . - \name
+ELF     .size   FUNC_PREFIX\name, . - FUNC_PREFIX\name
         .endfunc
         .purgem endfunc
     .endm
@@ -55,11 +61,11 @@ ELF     .size   \name, . - \name
         .align 2
 
     .if \export
-        .global \name
+        .global FUNC_PREFIX\name
     .endif
-ELF     .type   \name, STT_FUNC
-        .func   \name
-\name:
+ELF     .type   FUNC_PREFIX\name, STT_FUNC
+        .func   FUNC_PREFIX\name
+FUNC_PREFIX\name:
 .endm
 
 func    png_read_filter_row_sub4_neon, export=1
-- 
2.46.0

