From c38a0b50e6be3719acf422b80584ad4ac4cce038 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Fri, 23 Aug 2024 21:27:35 +0000
Subject: [PATCH 1/2] cmake: Force enable pkg-config modules everywhere

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f308590..53d461c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1031,7 +1031,8 @@ endif()
 # We use the same files like ./configure, so we have to set its vars.
 # Only do this on Windows for Cygwin - the files don't make much sense
 # outside of a UNIX look-alike.
-if(NOT WIN32 OR CYGWIN OR MINGW)
+# (PS: This is not correct on Meson-compliant systems! --amyspark)
+if(TRUE)
   set(prefix "${CMAKE_INSTALL_PREFIX}")
   set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
   set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
@@ -1101,7 +1102,7 @@ if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
   install(FILES png.5
           DESTINATION "${CMAKE_INSTALL_MANDIR}/man5")
   # Install the pkg-config files.
-  if(NOT CMAKE_HOST_WIN32 OR CYGWIN OR MINGW)
+  if(TRUE)
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng.pc"
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
     install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng-config"
-- 
2.53.0

