From 22f76fd53f8edcef6b10c678a60ef59105f69cf4 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 29 Oct 2013 02:10:06 +0100
Subject: [PATCH] Don't override environment AR

---
 lib/Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Makefile.in b/lib/Makefile.in
index b7b96ee..083d972 100644
--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -41,7 +41,9 @@ CXXCPP = @CXXCPP@
 # Both C and C++ compiler
 OBJEXT = @OBJEXT@
 # Other
+ifndef AR
 AR = ar
+endif
 AR_FLAGS = rc
 RANLIB = @RANLIB@
 MV = mv
-- 
1.8.1.2

