From cd8a2872bf58ec7db0335cce3ca2505d8fc55152 Mon Sep 17 00:00:00 2001
From: Olivier Crete <olivier.crete@collabora.com>
Date: Fri, 27 Apr 2018 17:45:47 -0400
Subject: [PATCH] Disable -fembed-bitcode incompatible argument

---
 configure    |  2 +-
 3 files changed, 19 insertions(+), 40 deletions(-)

diff --git a/configure b/configure
index 9d1586c..7ae8ee4 100755
--- a/configure
+++ b/configure
@@ -1425,7 +1425,7 @@ if [ "$shared" = "yes" ]; then
     elif [ "$SYS" = "MACOSX" ]; then
         echo "SOSUFFIX=dylib" >> config.mak
         echo "SONAME=libx264.$API.dylib" >> config.mak
-        echo "SOFLAGS=-shared -dynamiclib -Wl,-single_module -Wl,-read_only_relocs,suppress -install_name \$(DESTDIR)\$(libdir)/\$(SONAME) $SOFLAGS" >> config.mak
+        echo "SOFLAGS=-shared -dynamiclib -Wl,-single_module -install_name \$(DESTDIR)\$(libdir)/\$(SONAME) $SOFLAGS" >> config.mak
     elif [ "$SYS" = "SunOS" ]; then
         echo "SOSUFFIX=so" >> config.mak
         echo "SONAME=libx264.so.$API" >> config.mak
-- 
2.15.1 (Apple Git-101)

