From 1b0fa783a8199790abdafc49cfa0d71dc398c1e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Cerveau?= <scerveau@igalia.com>
Date: Fri, 16 Feb 2024 11:37:42 +0100
Subject: [PATCH] configure: Force pkgconfig file to have relative values

Use rel_libdir
---
 configure | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 3cf63e0..8308fd5 100755
--- a/configure
+++ b/configure
@@ -534,6 +534,7 @@ CC="${CC-${cross_prefix}gcc}"
 STRIP="${STRIP-${cross_prefix}strip}"
 INSTALL="${INSTALL-install}"
 PKGCONFIG="${PKGCONFIG-${cross_prefix}pkg-config}"
+rel_libdir="${libdir#$prefix/}"
 
 # ar and ranlib doesn't load the LTO plugin by default, prefer the gcc-prefixed wrappers which does.
 if ${cross_prefix}gcc-ar --version >/dev/null 2>&1; then
@@ -1530,8 +1531,8 @@ echo "CLI_LIBX264 = $CLI_LIBX264" >> config.mak
 cat > x264.pc << EOF
 prefix=$prefix
 exec_prefix=$exec_prefix
-libdir=$libdir
-includedir=$includedir
+libdir=\${prefix}/$rel_libdir
+includedir=\${prefix}/include
 
 Name: x264
 Description: H.264 (MPEG4 AVC) encoder library
-- 
2.34.1

