From 50d5d6a4156902c986d9d60e92f9e211ff5408c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Sat, 13 Sep 2014 20:10:15 +0200
Subject: [PATCH] Don't use -falign-loops with clang, it does not exist

---
 configure | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure b/configure
index 0177cfc..d5869a5 100755
--- a/configure
+++ b/configure
@@ -467,7 +467,6 @@ case $host_os in
         ;;
     darwin*)
         SYS="MACOSX"
-        CFLAGS="$CFLAGS -falign-loops=16"
         libm="-lm"
         if [ "$pic" = "no" ]; then
             cc_check "" -mdynamic-no-pic && CFLAGS="$CFLAGS -mdynamic-no-pic"
-- 
1.9.3 (Apple Git-50)

